{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "alike-seminar",
   "metadata": {
    "id": "alike-seminar"
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "# Para que no trunque las filas pandas\n",
    "pd.set_option('display.max_rows', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "TMMPZD4FQXY0",
   "metadata": {
    "id": "TMMPZD4FQXY0"
   },
   "outputs": [],
   "source": [
    "# esto es para google drive\n",
    "from google.colab import drive\n",
    "drive.mount('/content/drive')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "norwegian-waste",
   "metadata": {
    "id": "norwegian-waste"
   },
   "outputs": [],
   "source": [
    "#df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Tabla_R.csv')\n",
    "\n",
    "# Las columnas son \n",
    "# 1 -> First Author/Year of  Publication/ Country\n",
    "# 2 -> Database\n",
    "# 3 -> Initial Scaffold (ID)\n",
    "# 4 -> Designed Protein\n",
    "# 5 -> ML Model\n",
    "# 6 -> Software / Sever\n",
    "# 7 -> Programming Language / Platform\n",
    "# 8 -> License\n",
    "# 9 -> Quality (%)\n",
    "# 10 -> Machine Learning\n",
    "# 11 -> Protein Application\n",
    "df=pd.read_csv('Tabla_1_Total_.csv', names=['1','2','3','4','5','6','7','8','9','10','11'], encoding='utf8',header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a19f63f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>First Author/Year of  Publication/ Country</td>\n",
       "      <td>Database</td>\n",
       "      <td>Initial Scaffold (ID)</td>\n",
       "      <td>Designed Protein</td>\n",
       "      <td>ML Model</td>\n",
       "      <td>Software / Sever</td>\n",
       "      <td>Programming Language / Platform</td>\n",
       "      <td>License</td>\n",
       "      <td>Quality (%)</td>\n",
       "      <td>Machine Learning</td>\n",
       "      <td>Protein Application</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Burak T., 2021, Turkey</td>\n",
       "      <td>UniProt</td>\n",
       "      <td>Protein sequence from 60 different families</td>\n",
       "      <td>Protein family classification/identification</td>\n",
       "      <td>FIBHASH (based on Fibonacci numbers), RNN, LST...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>70</td>\n",
       "      <td>Supervised Learning:\\nClassification</td>\n",
       "      <td>Protein classification</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Song J., 2021, \\nChina</td>\n",
       "      <td>PDB\\nSwiss-Prot</td>\n",
       "      <td>ATP-binding proteins</td>\n",
       "      <td>Protein–ATP Binding Residues</td>\n",
       "      <td>CNN\\nLightGBM</td>\n",
       "      <td>TensorFlow, Keras\\nhttps://github.com/tlsjz/AT...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>80</td>\n",
       "      <td>Supervised Learning:\\nRegression\\nPrediction\\n...</td>\n",
       "      <td>Prediction of Protein–ATP Binding Residues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Verma N., 2021, \\nUS</td>\n",
       "      <td>DrugBank\\nMatador\\nPDB</td>\n",
       "      <td>Human, \\nC. Elegans</td>\n",
       "      <td>Protein ligand interactions</td>\n",
       "      <td>DNN</td>\n",
       "      <td>GitHub (https://github.com/ekraka/SSnet)</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning:\\nPrediction</td>\n",
       "      <td>Protein ligand interaction prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Li H., 2020, \\nFrance, \\nHong Kong,</td>\n",
       "      <td>PDB,\\nPubChem,\\nZINC,\\nChEMB,L\\nBindingDB,\\nHTS</td>\n",
       "      <td>Chemical\\nEstrogen receptor α (ERα)\\nAnaplasti...</td>\n",
       "      <td>Protein-ligand complex</td>\n",
       "      <td>RF, BRT, kNN, NN, SVM,  GBDT,\\nmulti-task\\nDNN...</td>\n",
       "      <td>Descriptor data bank\\nODDT\\nBINANA\\nRF-Score-v...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised learning models</td>\n",
       "      <td>Physicochemical properties</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Kwon Y., \\n2020, Korea</td>\n",
       "      <td>PDBind-2016</td>\n",
       "      <td>VEGFR2 kinase domain And adenosine deaminase</td>\n",
       "      <td>Prediction of affinity-binding of a protein-li...</td>\n",
       "      <td>3D CNN</td>\n",
       "      <td>Keras, Tensorflow</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>85</td>\n",
       "      <td>Supervised Learning:\\nPrediction</td>\n",
       "      <td>Protein affinity-binding prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Wang M.., 2020, US</td>\n",
       "      <td>\\nSKEMPI 1.0, 2.0 dataSet\\nAB-Bind S645 dataset</td>\n",
       "      <td>Protein-protein complexes</td>\n",
       "      <td>Protein-ligand binding affinity predictions</td>\n",
       "      <td>site-specific persistent homology, CNN, GBT</td>\n",
       "      <td>TopNetTree, Keras\\nhttps://doi.org/10.24433/CO...</td>\n",
       "      <td>Matlab, Java, Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>90</td>\n",
       "      <td>Supervised Learning:\\nPrediction</td>\n",
       "      <td>Protein-protein binding affinity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Qin Z., \\n2020, US</td>\n",
       "      <td>PDB</td>\n",
       "      <td>Phi–psi angle and sequence of natural protein,...</td>\n",
       "      <td>Protein design of fold alpha-helical structure</td>\n",
       "      <td>Multi-scale Neighborhood-based Neural Network ...</td>\n",
       "      <td>Tensorflow\\nhttps://github.com/IBM/mnnn.</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>95</td>\n",
       "      <td>Supervised Learning:\\nPrediction\\nRegression</td>\n",
       "      <td>Protein design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Zheng W., 2020, \\nChina, US</td>\n",
       "      <td>SCOPe2.07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Protein domain boundaries</td>\n",
       "      <td>FUpred</td>\n",
       "      <td>https://zhanglab.ccmb.med.umich.edu/FUpred/</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Open\\nSource</td>\n",
       "      <td>60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Protein domain identification</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Yu C., 2020, \\nTaiwan, US</td>\n",
       "      <td>PDB</td>\n",
       "      <td>α-helix rich proteins</td>\n",
       "      <td>De novo protein design</td>\n",
       "      <td>RNN, LSTM</td>\n",
       "      <td>TensorFlow, \\nhttps://github.com/tensorflow/ma...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>90</td>\n",
       "      <td>Supervised Learning;\\nUnsupervised Learning;\\n...</td>\n",
       "      <td>Protein Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Yang H., 2020, \\nChina</td>\n",
       "      <td>SCOPe 2.07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Contact map protein prediction</td>\n",
       "      <td>GAN</td>\n",
       "      <td>Keras, Tensorflow\\nhttps://github.com/melissay...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open\\nSource</td>\n",
       "      <td>70</td>\n",
       "      <td>Supervised Learning: \\nRegression</td>\n",
       "      <td>Contact map\\nprediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Wang X., \\n2020, \\nChina</td>\n",
       "      <td>KIBA, Davis dataset</td>\n",
       "      <td>Kinase protein family</td>\n",
       "      <td>Predicte Drug-target binding affinity</td>\n",
       "      <td>CNN\\nGCN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60</td>\n",
       "      <td>Supervised Learning, Semi-Supervised Learning:...</td>\n",
       "      <td>Drug-target binding affinity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Maia E., 2020, \\nBrazil</td>\n",
       "      <td>Multiple databases</td>\n",
       "      <td>structure-based virtual screening (SBVS)</td>\n",
       "      <td>Drug development</td>\n",
       "      <td>ANN;\\nSVM\\nBT\\nDT\\nkNN\\nEnsemble methods\\nGA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Multiple languages</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60</td>\n",
       "      <td>Supervised Learning;\\nUnsupervised Learning</td>\n",
       "      <td>Drug development o design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Shroff R., 2020, \\nUS</td>\n",
       "      <td>PDB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>amino acids association guide mutation</td>\n",
       "      <td>3D CNN</td>\n",
       "      <td>Theano\\nwww.\\nmutcompute.com</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open \\nSource</td>\n",
       "      <td>70</td>\n",
       "      <td>Supervised Learning:\\nClass\\nPrediction</td>\n",
       "      <td>Microenvironment mutation identification</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Mahmoud A., \\n2020, Switzerland, US</td>\n",
       "      <td>PDB</td>\n",
       "      <td>HIV-1 protease, dihydrofolate reductase</td>\n",
       "      <td>Hydration\\nsite occupancy and thermodynamics p...</td>\n",
       "      <td>CNN</td>\n",
       "      <td>https://hub.docker.com/r/lilllab/\\nwatsite3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Open\\nSource</td>\n",
       "      <td>65</td>\n",
       "      <td>Supervised Learning: Regression\\nClassification</td>\n",
       "      <td>Protein-ligand binding prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Wardah W., 2020, \\nAustralia, Fiji, Japan, US</td>\n",
       "      <td>BioLiP</td>\n",
       "      <td>Positive (binding) or negative (non-binding), ...</td>\n",
       "      <td>Prediction Protein-peptide binding sites</td>\n",
       "      <td>CNN</td>\n",
       "      <td>PyTorch, https://github.com/WafaaWardah/Visual</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning, Prediction</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Liang M., \\n2020, \\nChina</td>\n",
       "      <td>PDB</td>\n",
       "      <td>Relative angle of (C − Cα − C) principal plane</td>\n",
       "      <td>Enzymatic function prediction</td>\n",
       "      <td>RNN, LSTM</td>\n",
       "      <td>TensorFlow</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>90</td>\n",
       "      <td>Supervised Learning Prediction.</td>\n",
       "      <td>Protein function prediction, Function ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Feger G., 2020,\\nCzech Republic, France</td>\n",
       "      <td>PDB</td>\n",
       "      <td>Peptide amphiphiles\\nscaffolds</td>\n",
       "      <td>Amphiphilic peptide scaffolds design</td>\n",
       "      <td>SVM, RF</td>\n",
       "      <td>SasFit</td>\n",
       "      <td>C</td>\n",
       "      <td>Open source</td>\n",
       "      <td>60</td>\n",
       "      <td>Supervised Learning Prediction</td>\n",
       "      <td>Protein design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Dai W., China, 2020</td>\n",
       "      <td>Reactome DB &amp; InBio Map DB</td>\n",
       "      <td>Human essential gene</td>\n",
       "      <td>Predict human essential genes</td>\n",
       "      <td>SVM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "      <td>Supervised Learning, Classifier</td>\n",
       "      <td>Human genes prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Bond P., 2020, \\nUK</td>\n",
       "      <td>PDB</td>\n",
       "      <td>Only residues\\nwith side chains longer than be...</td>\n",
       "      <td>Prediction the correctness of protein residues</td>\n",
       "      <td>NN, MLP</td>\n",
       "      <td>CCP4</td>\n",
       "      <td>C++, \\nPython</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>60</td>\n",
       "      <td>Supervised Learning, Prediction</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Nie J., 2020, Singapore, Taiwan</td>\n",
       "      <td>UniProt</td>\n",
       "      <td>acetyl-lysine’ (S1), ‘crotonyl-lysine’ (S2),\\n...</td>\n",
       "      <td>Identification of Lysine PTM sites</td>\n",
       "      <td>RF,\\n SVM, MNB,\\nLR, Max Entropy, KNN,\\nCNN,\\nMLP</td>\n",
       "      <td>Tensorflo, https://github.com/khanhlee/\\nlysin...</td>\n",
       "      <td>Python</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Function ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Savojardo C., 2020, \\nItaly</td>\n",
       "      <td>UniProtKB\\nGOA,\\nDeepMitoDB</td>\n",
       "      <td>Human, mouse, fly, yeast and Arabidopsis thaliana</td>\n",
       "      <td>Protein sub-mitochondrial localization</td>\n",
       "      <td>DeepMito, \\n1D-CNN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised learning: Prediction</td>\n",
       "      <td>Function ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Du Z., 2020, \\nChina, Russia, US</td>\n",
       "      <td>CAFA3, SwissProt</td>\n",
       "      <td>Human, \\nC. Elegans</td>\n",
       "      <td>Automated function prediction</td>\n",
       "      <td>NLP, CNN</td>\n",
       "      <td>Keras, TensorFlow</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>70</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Li C., 2019, \\nChina</td>\n",
       "      <td>LE dataset from SCOP</td>\n",
       "      <td>Multiple superfamilies</td>\n",
       "      <td>Detect the structural\\nmotifs related with the...</td>\n",
       "      <td>MotifCNN and MotifDCNN\\nSVM \\nCNN</td>\n",
       "      <td>TensorFlow</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Fold ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Lin J., 2019, China</td>\n",
       "      <td>DrugBank</td>\n",
       "      <td>druggable proteins  and non-druggable proteins</td>\n",
       "      <td>Drug target prediction</td>\n",
       "      <td>SVM, GA</td>\n",
       "      <td>https://github.com/QUST-AIBBDRC/GA-Bagging-SVM</td>\n",
       "      <td>Matlab</td>\n",
       "      <td>MathWorks</td>\n",
       "      <td>90</td>\n",
       "      <td>Supervised Learning:\\nPrediction</td>\n",
       "      <td>Drug design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Fang C., 2019, US</td>\n",
       "      <td>UniProt</td>\n",
       "      <td>Proteins from datasets BT426 and BT6376, conta...</td>\n",
       "      <td>Beta-turn prediction</td>\n",
       "      <td>HMM, CNN, deep dense inception network (DeepDIN</td>\n",
       "      <td>Tensorflow, Keras \\nhttp://dslsrv8.cs.missouri...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>90</td>\n",
       "      <td>Supervised Learning:\\nClassification</td>\n",
       "      <td>Protein design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Wan C., 2019, \\nUK</td>\n",
       "      <td>UniProtKB/Swiss-Prot</td>\n",
       "      <td>Human proteins</td>\n",
       "      <td>Function prediction</td>\n",
       "      <td>Deep mahout neural network (DMNN), SVM</td>\n",
       "      <td>Keras,\\nhttps://github.com/psipred/STRING2GO</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>80</td>\n",
       "      <td>Supervised Learning: \\nPrediction\\nClassification</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Luo X., 2019, \\nChina</td>\n",
       "      <td>Transfac</td>\n",
       "      <td>DNA sequences</td>\n",
       "      <td>predicting DNA–protein binding</td>\n",
       "      <td>CNN</td>\n",
       "      <td>Keras, Tensorflow\\nhttps://github.com/gao-lab/...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>70</td>\n",
       "      <td>Supervised Learning:\\nRegression\\nPrediction</td>\n",
       "      <td>Protein binding prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Mirabello C., 2019, \\nSweden</td>\n",
       "      <td>PDB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Method prediction</td>\n",
       "      <td>DNN</td>\n",
       "      <td>Keras, TensorFlow\\nhttps://bitbucket.org/clami...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>70</td>\n",
       "      <td>Supervised Learning:\\nPrediction</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Savojardo C., 2019, \\nItaly</td>\n",
       "      <td>UniprotKB/Swiss-Prot</td>\n",
       "      <td>mitochondrial proteins</td>\n",
       "      <td>sub-mitochondrial cellular\\nlocalization.</td>\n",
       "      <td>CNN</td>\n",
       "      <td>http://busca.biocomp.unibo.it/deepmito</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: \\nRegression</td>\n",
       "      <td>protein\\nsub-mitochondrial site prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Rifaioglu A., 2019, \\nTurkey, UK</td>\n",
       "      <td>UniProtKB/Swiss-Prot</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GO terms prediction</td>\n",
       "      <td>DNN</td>\n",
       "      <td>Tensorflow,\\nhttps://github.com/cansyl/DEEPred</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open\\nSource</td>\n",
       "      <td>70</td>\n",
       "      <td>Supervised Learning:\\nRegression</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Han X., \\n2019, Singapore, US</td>\n",
       "      <td>eSol</td>\n",
       "      <td>ell-free protein expression from E.\\ncoli</td>\n",
       "      <td>Protein solubility</td>\n",
       "      <td>DNN\\nGAN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60</td>\n",
       "      <td>Supervised Learning:\\nRegression\\nPrediction</td>\n",
       "      <td>Protein solubility prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Zhao Z., 2019, \\nChina</td>\n",
       "      <td>Monomers and dimers from author</td>\n",
       "      <td>Monomers and dimers from author</td>\n",
       "      <td>Protein-protein interaction</td>\n",
       "      <td>LSTM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60</td>\n",
       "      <td>Supervised Learning, Unsupervised Learning: \\n...</td>\n",
       "      <td>Interface residue\\npairs prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Long S., 2019, \\nChina</td>\n",
       "      <td>Jpred dataset\\ncullpdb dataset\\nUniRef90\\nUniProt</td>\n",
       "      <td>Multiple superfamilies</td>\n",
       "      <td>Protein secondary structure prediction</td>\n",
       "      <td>CNN</td>\n",
       "      <td>TensorFlow\\nN/A</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open\\nSource</td>\n",
       "      <td>60</td>\n",
       "      <td>Supervised Learning;\\nUnsupervised Learning;\\n...</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Zhang Y., China, 2019</td>\n",
       "      <td>PDB</td>\n",
       "      <td>PDNA-543, PDNA-224 and PDNA-316</td>\n",
       "      <td>Identification of DNA–protein binding site</td>\n",
       "      <td>Adaptive Synthetic Sampling (ADASYN) approach\\...</td>\n",
       "      <td>Theano</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>85</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Function ID and Fold ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Kaleel M., 2019, \\nIreland</td>\n",
       "      <td>PDB</td>\n",
       "      <td>amino acids are categorised into four classes ...</td>\n",
       "      <td>Prediction of relative solvent accessibility</td>\n",
       "      <td>BRNN, CNN</td>\n",
       "      <td>http://distilldeep.ucd.ie/paleale/</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>90</td>\n",
       "      <td>Supervised Learning, Prediction</td>\n",
       "      <td>Protein relative solvent accessibility  predic...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Schantz M., 2019, \\nArgentina, \\nDenmark, \\nMa...</td>\n",
       "      <td>PDB, PISCES</td>\n",
       "      <td>Crystal structures</td>\n",
       "      <td>Prediction of protein structural features</td>\n",
       "      <td>CNN, LSTM</td>\n",
       "      <td>Keras</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning, Prediction</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Luo L., \\n2019, \\nChina</td>\n",
       "      <td>BioCreative II,\\nBioCreative III,\\nBioCreative...</td>\n",
       "      <td>PPI protein articles</td>\n",
       "      <td>Protein-protein interaction</td>\n",
       "      <td>Knowledge-enriched Self-Attention Convolutiona...</td>\n",
       "      <td>Keras</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>50</td>\n",
       "      <td>Classification of articles</td>\n",
       "      <td>Physicochemical properties</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Haberal I., 2019, Norway, Turkey</td>\n",
       "      <td>PDB</td>\n",
       "      <td>Metal-binding of histidines &amp; Cysteines amino ...</td>\n",
       "      <td>Prediction of metal-binding in proteins</td>\n",
       "      <td>CNN, LSTM, RNN</td>\n",
       "      <td>Keras, TensorFlow</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning: Prediction</td>\n",
       "      <td>Protein site prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Heinzinger M., 2019, \\nGermany</td>\n",
       "      <td>UniProt, PDB</td>\n",
       "      <td>TS115\\nCB513\\nCASP12</td>\n",
       "      <td>Protein sequence representation</td>\n",
       "      <td>ELMo\\nCharCNN  LSTMs</td>\n",
       "      <td>Pytorch, https://embed.protein.properties/</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>80</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Fold ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Zhang D., 2019, \\nUS</td>\n",
       "      <td>DIP, HPRD,\\nUniProt</td>\n",
       "      <td>D. melanogaster, S. cerevisiae,\\nE. coli, C. e...</td>\n",
       "      <td>Protein-protein interactions and protein famil...</td>\n",
       "      <td>Multimodal DNN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Physicochemical properties</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Fang C., 2019, \\nChina, Japan</td>\n",
       "      <td>PDB</td>\n",
       "      <td>MoRF-containing membrane protein chains</td>\n",
       "      <td>Molecular recognition features MoRFs prediction</td>\n",
       "      <td>DCNN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Function ID and Fold ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Cui Y., \\n2019, \\nChina\\n</td>\n",
       "      <td>BioLip</td>\n",
       "      <td>Fourteen binding residues</td>\n",
       "      <td>Protein-ligand binding residue prediction</td>\n",
       "      <td>DCNN</td>\n",
       "      <td>TensorFlow, https://github.com/yfCuiFaith/Deep...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning: Prediction</td>\n",
       "      <td>Protein site prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Taherzadeh G., 2019, Australia, US</td>\n",
       "      <td>Uniprot, dbPTM, Uniprep, UnicarKB, GlycoProtDB</td>\n",
       "      <td>glycoprotein</td>\n",
       "      <td>N- and O-linked\\nglycosylation</td>\n",
       "      <td>DNN SVM</td>\n",
       "      <td>TensorFlow,\\nhttps://sparks-lab.org/server/spr...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>80</td>\n",
       "      <td>Supervised Learning: \\nRegression\\nPrediction</td>\n",
       "      <td>Glycosylation\\nsite identification</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>Guo Y., 2019, US</td>\n",
       "      <td>CB513, CASP10, CASP11</td>\n",
       "      <td>Protein sequences</td>\n",
       "      <td>Protein secondary structure</td>\n",
       "      <td>ACNN, BLSTM</td>\n",
       "      <td>Keras, Tensorflow, https://\\ngithub.com/GYBTA/...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>80</td>\n",
       "      <td>Supervised Learning: \\nPrediction\\nClassification</td>\n",
       "      <td>Protein secondary structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Yang R., \\n2019, China</td>\n",
       "      <td>Datasets A(CPLM),B,C</td>\n",
       "      <td>Proteins and reducing sugars</td>\n",
       "      <td>Glycation product prediction</td>\n",
       "      <td>RNN, CNN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Glycation site predictor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Zheng W., 2019, US</td>\n",
       "      <td>CASP13</td>\n",
       "      <td>Query sequence profiles</td>\n",
       "      <td>Automated structure prediction pipeline</td>\n",
       "      <td>ZhangServer and QUARK pipelines</td>\n",
       "      <td>Zhang and Quark server</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>85</td>\n",
       "      <td>Supervised Learning: Classification\\nRegression</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Wang D., 2019, China</td>\n",
       "      <td>CASP11, 12</td>\n",
       "      <td>Caspase 14</td>\n",
       "      <td>Protein structure refinement</td>\n",
       "      <td>Monte Carlo, multi-objective particle swarm op...</td>\n",
       "      <td>AIR 2.0 www.csbio.sjtu.edu.cn/bioinf/AIR/</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>95</td>\n",
       "      <td>Supervised Learning: Optimization</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Fu H.,\\n2019, \\nChina</td>\n",
       "      <td>PLMD</td>\n",
       "      <td>Sequences and physicochemical properties of pr...</td>\n",
       "      <td>Predict Lysine ubiquitination sites in large-s...</td>\n",
       "      <td>CNN  DL   DeepUbi</td>\n",
       "      <td>TensorFlow,    DeepUbi: https://github.com/Sun...</td>\n",
       "      <td>Python, Matlab,\\nLinux</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein site prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>Pagès G., \\n2019, \\nFrance</td>\n",
       "      <td>CASP</td>\n",
       "      <td>Model QA</td>\n",
       "      <td>Protein model quality assessment</td>\n",
       "      <td>3D CNN</td>\n",
       "      <td>TensorFlow ,  Ornate  https://team.inria.fr/na...</td>\n",
       "      <td>C++,  \\nPython</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>85</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Model protein prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Torng  W., \\n2019,  \\nUS</td>\n",
       "      <td>PROSITE     NOS dataset</td>\n",
       "      <td>Protein structure as 3D images</td>\n",
       "      <td>Protein functional site detection</td>\n",
       "      <td>DL  3D CNN  SVM</td>\n",
       "      <td>N/A    https://simtk.org/projects/fscnn</td>\n",
       "      <td>Python</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>Yu C., \\n2019, \\nUS</td>\n",
       "      <td>PDB</td>\n",
       "      <td>194l (lysozyme), 107m (myoglobin), 6cgz (β-bar...</td>\n",
       "      <td>Generation of  audible sound  from amino acids...</td>\n",
       "      <td>RNN  LSTM</td>\n",
       "      <td>Magenta TensorFlow,  Melody RNN</td>\n",
       "      <td>Java, \\nPython</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Protein sequence prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Huang L., 2018, US</td>\n",
       "      <td>DIP, HPRD</td>\n",
       "      <td>PPI network graph</td>\n",
       "      <td>Protein-Protein interaction</td>\n",
       "      <td>Evolution neural network based regularized Lap...</td>\n",
       "      <td>TensorFlow, https://www.eecis.udel.edu/~lliao/...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning:\\nPrediction</td>\n",
       "      <td>Protein-protein interaction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Hanson J., Australia, China, \\n2018</td>\n",
       "      <td>PISCES\\nCASP12\\nPDB</td>\n",
       "      <td>5N5EA \\n6FI2A\\n6FQ3A</td>\n",
       "      <td>Sequence-based prediction of one dimensional s...</td>\n",
       "      <td>CNN, 2D-BRLSTM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Function ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>Hanson J., 2018, Australia, China</td>\n",
       "      <td>PDB\\nUniProt</td>\n",
       "      <td>Primary amino-acid-sequence, proteins from CASP12</td>\n",
       "      <td>Protein contact map prediction</td>\n",
       "      <td>CNN, 2D bidirectional recurrent LSTM (BRLSTM)</td>\n",
       "      <td>http://sparks-lab.org/jack/server/SPOTContact/</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>95</td>\n",
       "      <td>Supervised Learning:\\nPrediction</td>\n",
       "      <td>Protein contact-map prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>Zhao X., 2018, \\nChina</td>\n",
       "      <td>PLMD</td>\n",
       "      <td>Lysine</td>\n",
       "      <td>Lysine acetylation sites</td>\n",
       "      <td>CNN\\nDNN</td>\n",
       "      <td>Keras, Theano,\\nhttps://github.com/jiagenlee/D...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>80</td>\n",
       "      <td>Supervised Learning:\\nRegression\\nClassificati...</td>\n",
       "      <td>Acetylation site prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>O’Connell J., 2018, Australia, China, US</td>\n",
       "      <td>SPIN dataset</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sequences profile compatible</td>\n",
       "      <td>DNN</td>\n",
       "      <td>http://sparks-lab.org.\\n\\nSPIN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>65</td>\n",
       "      <td>Supervised Learning: \\nPrediction</td>\n",
       "      <td>Protein  sequence prediction\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>Xue L., \\n2018, \\nChina, US</td>\n",
       "      <td>Swiss-Prot, TrEMBL</td>\n",
       "      <td>Secretory protein</td>\n",
       "      <td>Protein sequence into T3SEs or non-T3SEs</td>\n",
       "      <td>CNN</td>\n",
       "      <td>Keras,\\nhttps://github.com/lje00006/DeepT3</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>60</td>\n",
       "      <td>Supervised Learning: \\nRegression\\nClassification</td>\n",
       "      <td>Protein classification</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>Feinberg E., \\n2018, \\nChina, US</td>\n",
       "      <td>PDB Bind 2007</td>\n",
       "      <td>Scaffold split for grouping ligands in common ...</td>\n",
       "      <td>Molecular Property Prediction</td>\n",
       "      <td>RNN DTNN &amp; MPNN</td>\n",
       "      <td>PyTorch, NumPy &amp; SciPy</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning, Prediction</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>Zhang L.,\\n2018, \\nChina</td>\n",
       "      <td>UniProt, \\nDIP</td>\n",
       "      <td>S. cerevisiae, H. sapiens, and M. musculus</td>\n",
       "      <td>Predicting Protein-Protein interactions</td>\n",
       "      <td>DNN, Adam Algorithm</td>\n",
       "      <td>TensorFlow</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning, Prediction</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Zhang B., 2018, \\nChina</td>\n",
       "      <td>PDB, PISCES, TR5534 Dataset</td>\n",
       "      <td>CASP10, 11, 12 and 13</td>\n",
       "      <td>Prediction of performance of protein</td>\n",
       "      <td>CNN, RNN, BRNN</td>\n",
       "      <td>Keras</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning, Prediction</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>Luo F., 2018, \\nChina</td>\n",
       "      <td>Phospho.ELM, PhosphositePlus, HPRD, dbPTM, SysPTM</td>\n",
       "      <td>Kinase protein family</td>\n",
       "      <td>Protein phosphorylation</td>\n",
       "      <td>CNN</td>\n",
       "      <td>https://\\ngithub.com/USTCHIlab/DeepPhos.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>60</td>\n",
       "      <td>Supervised Learning: \\nRegression\\nPredicition</td>\n",
       "      <td>Phosphorylation site predictor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>Long H.,  \\n2018, \\nChina, US</td>\n",
       "      <td>UniProt</td>\n",
       "      <td>PseAAC  Hydroxyproline &amp; hydroxylysine</td>\n",
       "      <td>Predicting hydroxylation sites</td>\n",
       "      <td>HDL  CNN  LSTM  RNN</td>\n",
       "      <td>MXNet ,   N/A</td>\n",
       "      <td>R</td>\n",
       "      <td>Apache  2.0</td>\n",
       "      <td>85</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>Frasca M., 2018, \\nItaly</td>\n",
       "      <td>STRING   GO</td>\n",
       "      <td>Organisms:   Homo sapiens (human)  S.cerevisia...</td>\n",
       "      <td>AFP (Automated Protein Function Prediction)</td>\n",
       "      <td>COSNet  ParCOSNet  HNN</td>\n",
       "      <td>COSNet,  ParCOSNet</td>\n",
       "      <td>C, C++, R, CUDA</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>75</td>\n",
       "      <td>Unsupervised Learning: Clustering</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>Hu B., \\n2018, \\nChina</td>\n",
       "      <td>DDI    SIDER  TWOSIDES   HPRD   Drug Bank  Off...</td>\n",
       "      <td>Semantic meta-paths  ADR</td>\n",
       "      <td>meta-path-based proximities ADR</td>\n",
       "      <td>SDHINE</td>\n",
       "      <td>TensorFlow, N/A</td>\n",
       "      <td>C, C++, \\nPython</td>\n",
       "      <td>Apache  2.0</td>\n",
       "      <td>65</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>Le N., \\n2018, \\nTaiwan</td>\n",
       "      <td>UniProt  GO</td>\n",
       "      <td>Rab GGT activity  Rab GDI activity  Rab GTPase...</td>\n",
       "      <td>Classify Rab protein molecules</td>\n",
       "      <td>CNN</td>\n",
       "      <td>Keras, Theano     DeepRab;  http://bio216.bioi...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>90</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Protein Classification</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>Sunseri J., 2018, \\nUS</td>\n",
       "      <td>D3R Grand challenge 3  Grand challenge 3</td>\n",
       "      <td>Input ligand  SMILES protein FASTA  CSAR</td>\n",
       "      <td>Cathepsin S  model ligand protein</td>\n",
       "      <td>CNN</td>\n",
       "      <td>Gnina ,  Caffe,  https://github.com/gnina</td>\n",
       "      <td>C++, \\nPython</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Protein model prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>Cuperus J., 2018, \\nUS</td>\n",
       "      <td>5′ UTR library of 50-nt-long random sequences</td>\n",
       "      <td>Yeast Saccharomyces cerevisiae</td>\n",
       "      <td>Predict protein expression</td>\n",
       "      <td>CNN</td>\n",
       "      <td>Keras, Theano,  https  ://github.com/Seeliglab...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>85</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Fold ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>Li H., \\n2018, \\nChina</td>\n",
       "      <td>HPRD  DIP   HIPPIE</td>\n",
       "      <td>Primary sequence  Escherichia coli,  Drosophil...</td>\n",
       "      <td>Prediction of protein interactions</td>\n",
       "      <td>DNN  CNN  LSTM</td>\n",
       "      <td>Keras, Theano,  TensorFlow , N/A</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>85</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>Popova M., 2018, \\nRussia, US</td>\n",
       "      <td>PHYSPROP  ChEMBL  KKB</td>\n",
       "      <td>SMILE string</td>\n",
       "      <td>Drug design (de novo design)</td>\n",
       "      <td>Stack-RNN  LSTM.  ReLeaSE</td>\n",
       "      <td>PyTorch, TensorFlow    ReLeaSE https://github....</td>\n",
       "      <td>Python, \\nCUDA</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>75</td>\n",
       "      <td>Reinforced Learning, Unsupervised Learning: Re...</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>Zafeiris D., 2018, \\nUK</td>\n",
       "      <td>GEO  Array Expression</td>\n",
       "      <td>Amyloid beta precursor protein, microtubule as...</td>\n",
       "      <td>Biomarker discovery for Alzheimer's disease</td>\n",
       "      <td>ANN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Enzyme Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>Zhao B., \\n2018, \\nUS</td>\n",
       "      <td>DisProt   PDB</td>\n",
       "      <td>Intrinsically disordered proteins (IDPs), intr...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ANN  DT</td>\n",
       "      <td>DisEMBL, IUPred, VSL2, DBann, and Espritz</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Intrinsically disordered protein prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>Hochuli J., \\n2018, \\nUS</td>\n",
       "      <td>PDB</td>\n",
       "      <td>Ligands SMILE Protein FASTA</td>\n",
       "      <td>Identify protein ligand scoring</td>\n",
       "      <td>CNN</td>\n",
       "      <td>Gnina , Caffe    Github.com/gnina</td>\n",
       "      <td>C++, Python</td>\n",
       "      <td>Open source</td>\n",
       "      <td>50</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein Scoring</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>Khurana S., 2018, Qatar, US</td>\n",
       "      <td>pepcDB database</td>\n",
       "      <td>k-mer structure and additional sequence and st...</td>\n",
       "      <td>Solubility prediction</td>\n",
       "      <td>CNN  DL  FFNN</td>\n",
       "      <td>PROSO II  https://zenodo.org/record/1162886#.X...</td>\n",
       "      <td>Python,\\n \\nLinux</td>\n",
       "      <td>Open  source</td>\n",
       "      <td>95</td>\n",
       "      <td>Unsupervised Learning: Clustering</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>Makrodimitris S., \\n2018,  Netherlands</td>\n",
       "      <td>Arabidopsis thaliana proteins</td>\n",
       "      <td>Arabidopsis thaliana protein</td>\n",
       "      <td>Protein function prediction</td>\n",
       "      <td>KNN  LSDR</td>\n",
       "      <td>SciPy     https://github.  com/stamakro/SSP-LSDR.</td>\n",
       "      <td>Python, Matlab   Bioinformatics toolbox</td>\n",
       "      <td>Open  source, Mathworks</td>\n",
       "      <td>80</td>\n",
       "      <td>Unsupervised Learning:  Clustering</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>Fang C., 2018, \\nUS</td>\n",
       "      <td>PDB\\nJPRED\\nCASP\\nCB513</td>\n",
       "      <td>Different super-families,\\nCASP10, 11, 12</td>\n",
       "      <td>Protein secondary structure prediction</td>\n",
       "      <td>Deep\\ninception-inside-inception (Deep3I) network</td>\n",
       "      <td>MUFOLD-SS\\nTensorFlow and Keras</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>80</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>Adhikari B., 2017,\\nUS</td>\n",
       "      <td>DNCON \\nDataset</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Contact map protein prediction</td>\n",
       "      <td>CNN</td>\n",
       "      <td>TensorFlow, Keras \\nhttp://sysbio.rnet.missour...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>65</td>\n",
       "      <td>Supervised Learning:\\nRegression\\nPredecition</td>\n",
       "      <td>Protein residue–residue contacts</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>Armenteros J., 2017, \\nDenmark</td>\n",
       "      <td>UniProt</td>\n",
       "      <td>Protein sequence,  Sequence information</td>\n",
       "      <td>Predict protein sub-cellular localization</td>\n",
       "      <td>CNN  RNN  BLSTM  FFNN  Attention models</td>\n",
       "      <td>Lasagne, Theano,   Deep Loc: http://www.cbs.dt...</td>\n",
       "      <td>Python</td>\n",
       "      <td>License MIT</td>\n",
       "      <td>90</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>Jimenez J., \\n2017, \\nSpain</td>\n",
       "      <td>scPDB</td>\n",
       "      <td>PDB ID File or PDB file</td>\n",
       "      <td>Predict protein-ligand-binding sites Drug design</td>\n",
       "      <td>3D DCNN</td>\n",
       "      <td>Keras, Theano    www.playmolecule.org</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>90</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>Wang S., \\n2017, \\nUS</td>\n",
       "      <td>Pfam   CASP   CAMEO</td>\n",
       "      <td>150 Pfam families 105 CASP11 test proteins  76...</td>\n",
       "      <td>5f5pH</td>\n",
       "      <td>UDNN  RNN 2</td>\n",
       "      <td>TensorFlow,  Theano    http://raptorx.uchicago...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Apache  2.0</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification Regression</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>Vang Y., \\n2017, \\nUS</td>\n",
       "      <td>IEDB  MHCBN SYFPEITHI</td>\n",
       "      <td>Human leukocyte antigen (HLA) complex</td>\n",
       "      <td>HLA class I-peptide binding prediction</td>\n",
       "      <td>CNN</td>\n",
       "      <td>Keras, Theano,   https://github.com/uci-cbcl/H...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>Müller A., 2017, Switzerland</td>\n",
       "      <td>ADAM  APD  DADP</td>\n",
       "      <td>Antimicrobial peptide   Amino acid sequences</td>\n",
       "      <td>Design of new peptides combinatorial de novo p...</td>\n",
       "      <td>RNN  LSTM</td>\n",
       "      <td>modlAMP  Python package    https://github.com/...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>Szalkai B., 2017,\\nHungary</td>\n",
       "      <td>Swiss-Prot, UniProt, \\nGO</td>\n",
       "      <td>Thyroid hormone,  phenol-containing\\ncompound,...</td>\n",
       "      <td>protein classification by amino-acid\\nsequence</td>\n",
       "      <td>DNN</td>\n",
       "      <td>TensorFlow</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>90</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein Classification</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>Szalkai B., \\n2017, \\nHungary</td>\n",
       "      <td>UniProt  GO</td>\n",
       "      <td>Classes.tre</td>\n",
       "      <td>Hierarchical Biological Sequence Classiﬁcation</td>\n",
       "      <td>ANN  DNN</td>\n",
       "      <td>SECLAF,  TensorFlow    https://pitgroup.org/se...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>85</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein Classification</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>Cao R, \\n2017, \\nUS</td>\n",
       "      <td>UniProt</td>\n",
       "      <td>Protein sequence</td>\n",
       "      <td>Protein function prediction</td>\n",
       "      <td>RNN</td>\n",
       "      <td>ProLanGO Model    N/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>Ragoza M., 2017,\\nUS</td>\n",
       "      <td>PDB  ChEMBL</td>\n",
       "      <td>Spatial and chemical  features of protein-liga...</td>\n",
       "      <td>Protein-ligand score for drug discovery</td>\n",
       "      <td>CNN  SGD</td>\n",
       "      <td>Gnina  Caffe    https://github.com/gnina</td>\n",
       "      <td>C++</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>85</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>Yeh C., \\n2017, \\nUK, US</td>\n",
       "      <td>JSON database:  centers of mass and geometric ...</td>\n",
       "      <td>Helical repeat proteins, Center of mass (CoM) ...</td>\n",
       "      <td>Designed helical repeat proteins (DHRs)</td>\n",
       "      <td>GA  multithreaded processing</td>\n",
       "      <td>ELFIN    https://github.com/joy13975/elfin</td>\n",
       "      <td>Python, C++, Matlab</td>\n",
       "      <td>Apache 2.0 open source    3-Clause BSD</td>\n",
       "      <td>90</td>\n",
       "      <td>Supervised Learning: Optimization</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>Al-Gharabli S., 2015, \\nJordan</td>\n",
       "      <td>PDB</td>\n",
       "      <td>Amino acid sequence hydrophobicity</td>\n",
       "      <td>Prediction of dihedral angles physiochemical p...</td>\n",
       "      <td>ANN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>Yang J., \\n2015, \\nChina, US</td>\n",
       "      <td>PDB  SPx dataset  PDBCYS dataset</td>\n",
       "      <td>Amino acid sequence</td>\n",
       "      <td>Structure  prediction of cysteine-rich proteins</td>\n",
       "      <td>HMM  SVR</td>\n",
       "      <td>CYSCON    http://www.csbio.sjtu.edu.cn/bioinf/...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>Simha R., 2015, \\nCanada, Germany, US</td>\n",
       "      <td>DBMLoc dataset</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Protein multi-location prediction</td>\n",
       "      <td>MDLoc  BN</td>\n",
       "      <td>MDLoc    http://www.eecis.udel.edu/compbio/mdloc.</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein site prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>Khan Z., \\n2014, \\nPakistan</td>\n",
       "      <td>BRENDA</td>\n",
       "      <td>Amino Acid sequence  and alkaline enzyme</td>\n",
       "      <td>Enzyme catalysis</td>\n",
       "      <td>KNN  SVM  MLP  PNN  DT</td>\n",
       "      <td>MATLAB  Bioweka  Weka</td>\n",
       "      <td>Java, Matlab</td>\n",
       "      <td>Open Source   MathWorks</td>\n",
       "      <td>50</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>Li Y., \\n2014, \\nUS</td>\n",
       "      <td>PDB</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>Designs of improved enzymes and enzymes with n...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>Murphy G., 2014, \\nUS</td>\n",
       "      <td>DND_4HB protein</td>\n",
       "      <td>DND_4HB protein</td>\n",
       "      <td>Design an up-down four-helix bundle.</td>\n",
       "      <td>Computational folding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>Folkman L., \\n2014, \\nAustralia</td>\n",
       "      <td>ProTherm</td>\n",
       "      <td>Protein sequence and amino acid substitution</td>\n",
       "      <td>Model designed for a specific type of mutation</td>\n",
       "      <td>SFFS  SVM  EASE-MM</td>\n",
       "      <td>EASE-MM  LISVM    http://www.ict.griffigr.edu....</td>\n",
       "      <td>Python, Linux</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Model Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>Li Z., \\n2014, \\nUS</td>\n",
       "      <td>PISCES</td>\n",
       "      <td>TL2282 dataset  TS500 dataset  TR1532 dataset</td>\n",
       "      <td>Sequence profile prediction</td>\n",
       "      <td>SPIN  NN</td>\n",
       "      <td>SPIN     http://sparks-lab.org</td>\n",
       "      <td>Python , Linux</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>85</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>Volpato V., \\n2013, \\nIreland</td>\n",
       "      <td>ENZYME UniProt</td>\n",
       "      <td>Oxidoreductase, transferase, hydrolase, lyase,...</td>\n",
       "      <td>Acid-residue frequency derived from multiple s...</td>\n",
       "      <td>N-to-1 Neural Networks</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>65</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>Wong K., \\n2013, \\nCanada, US, Saudi Arabia</td>\n",
       "      <td>Protein Binding Microarray data set</td>\n",
       "      <td>DNA sequence</td>\n",
       "      <td>DNA-motif discovery</td>\n",
       "      <td>Kmer-HMM</td>\n",
       "      <td>kmerHMM    http://www.cs.toronto.edu/wkc/  kme...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "      <td>Supervised Learning: Classification.  Unsuperv...</td>\n",
       "      <td>Model Discovery</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>Traoré S., 2013, \\nFrance</td>\n",
       "      <td>PDB</td>\n",
       "      <td>3D protein structure</td>\n",
       "      <td>Structure-based computational  protein design ...</td>\n",
       "      <td>CFN</td>\n",
       "      <td>CPD    http://genoweb.toulouse.inra.fr/tschiex...</td>\n",
       "      <td>Perl</td>\n",
       "      <td>Open  source</td>\n",
       "      <td>65</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>Qi Y., \\n2012, \\nUS</td>\n",
       "      <td>Standard benchmark, CB513  DSSP</td>\n",
       "      <td>PSI-BLAST  amino acid embedding</td>\n",
       "      <td>Prediction of local properties in proteins</td>\n",
       "      <td>DNN</td>\n",
       "      <td>Torch5</td>\n",
       "      <td>C</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>Eisenbeis S., \\n2012, \\nGermany</td>\n",
       "      <td>PDB</td>\n",
       "      <td>(βα)8-barrel and the flavodoxin-like fold,CheY...</td>\n",
       "      <td>Enzyme design</td>\n",
       "      <td>NaN</td>\n",
       "      <td>http://pubs.acs.org    Modeller, Rosetta</td>\n",
       "      <td>Python</td>\n",
       "      <td>IBM, Academic nonprofit freeware</td>\n",
       "      <td>75</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>Daniels N., 2012, \\nUS</td>\n",
       "      <td>SCOP</td>\n",
       "      <td>Protein sequence, 207 beta structural SCOP sup...</td>\n",
       "      <td>Detection for beta-structural proteins into th...</td>\n",
       "      <td>HMM  MRF</td>\n",
       "      <td>SMURFLite    http://smurf.cs.tufts.edu/smurflite/</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>65</td>\n",
       "      <td>Unsupervised Learning:  Clustering</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>Durrant J.,  2011, \\nUS</td>\n",
       "      <td>PDB  MOAD</td>\n",
       "      <td>Crystal structure data</td>\n",
       "      <td>Identiﬁcation of small-molecule ligands</td>\n",
       "      <td>ANN scoring function map</td>\n",
       "      <td>NNScore  2.0    http://www.nbcr.net/software/n...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>50</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein map prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>Ebina T.,  \\n2011, \\nJapan</td>\n",
       "      <td>DS-All  dataset</td>\n",
       "      <td>Protein sequence</td>\n",
       "      <td>Domain predictor</td>\n",
       "      <td>DROP  SVM  RF</td>\n",
       "      <td>DROP    http://web.tuat.ac.jp/~domserv/DROP.html</td>\n",
       "      <td>Bash script</td>\n",
       "      <td>Open  source</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>Ashkenazy H., \\n2011, \\nIsrael</td>\n",
       "      <td>PDB</td>\n",
       "      <td>3D protein structure</td>\n",
       "      <td>Contact map prediction</td>\n",
       "      <td>WMC</td>\n",
       "      <td>http://tau.ac.il/~haimash/WMC</td>\n",
       "      <td>Perl</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>45</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Protein map prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>Yang Y.,\\n2011, \\nUS</td>\n",
       "      <td>SPINE</td>\n",
       "      <td>Protein sequence</td>\n",
       "      <td>Single-method fold recognition</td>\n",
       "      <td>SPARKS-X  Algorithm</td>\n",
       "      <td>SPARKS-X    https://sparks-lab.org/server/spar...</td>\n",
       "      <td>Shell script</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>Latek D., \\n2010, \\nPoland</td>\n",
       "      <td>10 globular proteins 216 residues &amp; S100A1 pro...</td>\n",
       "      <td>10 globular proteins &amp; S100A1 protein</td>\n",
       "      <td>Predicted Nuclear Overhauser Effect signals on...</td>\n",
       "      <td>CABS  MC</td>\n",
       "      <td>CABS- NMR  toolkit    http://biocomp.chem.uw.e...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>70</td>\n",
       "      <td>Unsupervised Learning: Clustering</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>Zhao F., \\n2010, \\nUS</td>\n",
       "      <td>CASP</td>\n",
       "      <td>( PSSM) Position-specific scoring matrix  gene...</td>\n",
       "      <td>Protein folding</td>\n",
       "      <td>CNF  SVM</td>\n",
       "      <td>CNF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Fold ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>Briesemeister S., \\n2010, \\nGermany</td>\n",
       "      <td>UniProt</td>\n",
       "      <td>Protein sequence</td>\n",
       "      <td>Predict protein subcellular localization</td>\n",
       "      <td>NB</td>\n",
       "      <td>YLoc  Weka    www.multiloc.org/YLoc</td>\n",
       "      <td>Python, Java, Linux</td>\n",
       "      <td>Open  source</td>\n",
       "      <td>85</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein site prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>Lin G., \\n2010, \\nUS</td>\n",
       "      <td>PDB</td>\n",
       "      <td>Protein Folding Rates.  Predicting protein fol...</td>\n",
       "      <td>Protein folding kinetic rate and real-value fo...</td>\n",
       "      <td>SVM  SVR</td>\n",
       "      <td>SeqRate    http://casp.rnet.missouri.edu/fold_...</td>\n",
       "      <td>Java</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein map prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>Tian J., \\n2010, \\nChina, US</td>\n",
       "      <td>ProTherm  PDB</td>\n",
       "      <td>3D structures</td>\n",
       "      <td>Effect on single or multi-site mutation on pro...</td>\n",
       "      <td>RFR  SVM  RF</td>\n",
       "      <td>Prethermut     http://www.mobioinfor.cn/prethe...</td>\n",
       "      <td>R, Perl,  Linux</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>Bostan B., 2009,\\nUS</td>\n",
       "      <td>KEGG</td>\n",
       "      <td>Given a species proteome</td>\n",
       "      <td>Predict homologous signaling pathway</td>\n",
       "      <td>PSP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Model Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>Hong E.,\\n2009, \\nUS</td>\n",
       "      <td>Standard rotamer library  Expanded rotamer lib...</td>\n",
       "      <td>fn3: Derived from protein Fn3, tenth human fib...</td>\n",
       "      <td>Tenth human fibronectin, D44.1 and DI.3 antibo...</td>\n",
       "      <td>BroMAP</td>\n",
       "      <td>BroMAP</td>\n",
       "      <td>C++, Linux</td>\n",
       "      <td>Open   Source</td>\n",
       "      <td>100</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>Özen A., \\n2009, \\nTurkey</td>\n",
       "      <td>ProTherm</td>\n",
       "      <td>Structure-based features: amino acid substitut...</td>\n",
       "      <td>Single-site amino acid substitution</td>\n",
       "      <td>SVM  KNN  DT  SVR</td>\n",
       "      <td>MOSEK    http://www.prc.boun.edu.tr/appserv/pr...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Open  Source</td>\n",
       "      <td>85</td>\n",
       "      <td>Supervised Learning: Classification Regression</td>\n",
       "      <td>Model Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>Ebrahimpour A., \\n2008,  Malaysia</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>Geobacillus sp. strain</td>\n",
       "      <td>Lipase production Syncephalastrum racemosum, P...</td>\n",
       "      <td>ANN  FFNN  IBP  BBP    QP     GA   LM</td>\n",
       "      <td>CPC-X Software    N/A</td>\n",
       "      <td>Java</td>\n",
       "      <td>Neural Power version 2.5</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>Zhu X., \\n2008, \\nChina</td>\n",
       "      <td>PDB</td>\n",
       "      <td>223 scaffold proteins</td>\n",
       "      <td>Pocket residues of ribose-binding protein (2dr...</td>\n",
       "      <td>DGA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>65</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>Huang W., \\n2008, \\nTaiwan</td>\n",
       "      <td>UniProt  GO</td>\n",
       "      <td>SCL12, SCL16  Sequence-based, GO terms, protei...</td>\n",
       "      <td>Prediction method for predicting subcellular l...</td>\n",
       "      <td>GA  SVM  KNN</td>\n",
       "      <td>LIBSVM  ProlocGO    http://iclab.life.nctu.edu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein site prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>Wu S.,\\n2008, \\nUS</td>\n",
       "      <td>PDB</td>\n",
       "      <td>PDB protein sequence</td>\n",
       "      <td>Protein-contact predictor</td>\n",
       "      <td>MUSTER  SVM</td>\n",
       "      <td>MUSTER   http://zhang.bioinformatics.ku.edu/MU...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>Katzman S., 2008, \\nUS</td>\n",
       "      <td>PDB  PISCES</td>\n",
       "      <td>Amino acid sequence of a protein of unknown st...</td>\n",
       "      <td>Local structure prediction</td>\n",
       "      <td>Multi-layer NN</td>\n",
       "      <td>PREDICT-2ND    http://www.soe.ucsc.edu/∼karplu...</td>\n",
       "      <td>C++</td>\n",
       "      <td>Open source</td>\n",
       "      <td>80</td>\n",
       "      <td>Unsupervised Learning: Clustering</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>Shah R., \\n2008, \\nUS</td>\n",
       "      <td>D Dataset</td>\n",
       "      <td>Protein sequence</td>\n",
       "      <td>Homology detection</td>\n",
       "      <td>SVM</td>\n",
       "      <td>SVM-HUSTLE    http://www.sysbio.org/sysbio/net...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>70</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Function ID and Fold ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>Armstrong K., \\n2008, \\nUS</td>\n",
       "      <td>PDB</td>\n",
       "      <td>Protein sequence</td>\n",
       "      <td>Protein engineering space of foldable sequences</td>\n",
       "      <td>GR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C++</td>\n",
       "      <td>Open  source</td>\n",
       "      <td>50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fold ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>Shamim M., \\n2007, \\nIndia</td>\n",
       "      <td>D-B dataset  Ext. D-B dataset</td>\n",
       "      <td>Structural information of amino acid residue a...</td>\n",
       "      <td>Protein fold prediction</td>\n",
       "      <td>SVM</td>\n",
       "      <td>LIBSVM-library</td>\n",
       "      <td>C++, Java , Python    Windows ,Linux</td>\n",
       "      <td>Open   source</td>\n",
       "      <td>80</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Fold ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>Liao J., \\n2007, \\nUS</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>Proteinase K-catalyzed hydrolysis of the tetra...</td>\n",
       "      <td>Proteinase K variants</td>\n",
       "      <td>RR   Lasso PLSR SVMR LPSVMR  LPBoosR  MR  ORMR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Matlab</td>\n",
       "      <td>MathWorks</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification Regression</td>\n",
       "      <td>Protein design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>Raveh B., \\n2007, \\nIsrael</td>\n",
       "      <td>PDB</td>\n",
       "      <td>TIM-barrel fold 1YPI.  Whole β-sheets global s...</td>\n",
       "      <td>Existence of α-helices, parallel β-sheets, ant...</td>\n",
       "      <td>K-means Clustering</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Matlab</td>\n",
       "      <td>MathWorks</td>\n",
       "      <td>75</td>\n",
       "      <td>Unsupervised Learning: Clustering</td>\n",
       "      <td>Protein design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>Zimmermann O., \\n2006, \\nGermany</td>\n",
       "      <td>PDB</td>\n",
       "      <td>Protein sequence</td>\n",
       "      <td>Prediction of dihedral regions</td>\n",
       "      <td>SVM  C-SVM algorithm implementation</td>\n",
       "      <td>LIBSVM-library     DHPRED  http://www.fz-  jue...</td>\n",
       "      <td>C, Python, Linux,  Windows</td>\n",
       "      <td>Open  source</td>\n",
       "      <td>80</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>Sidhu A., \\n2006, \\nUK</td>\n",
       "      <td>Swiss-Prot</td>\n",
       "      <td>signal peptides and non-secretory proteins fro...</td>\n",
       "      <td>Predict signal peptide</td>\n",
       "      <td>BBFNN  NN  DT</td>\n",
       "      <td>BBFN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>Hung C., \\n2006, \\nTaiwan</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>Nucleocapsid (nsp1) of a coronavirus family</td>\n",
       "      <td>Predict protein functions</td>\n",
       "      <td>DFS  HMM  GA  AGCT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>Capriotti E., 2005, \\nItaly</td>\n",
       "      <td>ProTherm</td>\n",
       "      <td>Protein tertiary structure</td>\n",
       "      <td>Protein stability prediction</td>\n",
       "      <td>SVM</td>\n",
       "      <td>I-Mutant2.0     http://gpcr.biocomp.unibo.it/c...</td>\n",
       "      <td>Python</td>\n",
       "      <td>Open Source</td>\n",
       "      <td>75</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>Hu C.,\\n2004, \\nUS</td>\n",
       "      <td>WhatIF database UniProt</td>\n",
       "      <td>3D coarse-grained structure from protein seque...</td>\n",
       "      <td>Optimal non-linear scoring</td>\n",
       "      <td>SVM non-linear Gaussian kernel functions</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N/A.</td>\n",
       "      <td>65</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>Gutteridge A., 2003, \\nUK</td>\n",
       "      <td>PDB</td>\n",
       "      <td>Amino acid sequence of quinolate phosphoribosy...</td>\n",
       "      <td>Predict active site</td>\n",
       "      <td>NN  FFNN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "      <td>Unsupervised Learning:  Clustering</td>\n",
       "      <td>Protein function prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>Bindslev C., \\n2002, \\nDenmark</td>\n",
       "      <td>20 Patients with allergy to Macrozoarces ameri...</td>\n",
       "      <td>Macrozoarces americanus</td>\n",
       "      <td>Investigate potential allergenicity of Ice Str...</td>\n",
       "      <td>DT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>45</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Protein structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>Russ W., \\n2002, \\nUS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SH3 domain  GroEL minichaperone  WW domain pro...</td>\n",
       "      <td>Thermostable consensus phytase, 84.5 kDa protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>65</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Protein design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>Rossi A., \\n2001, \\nItaly</td>\n",
       "      <td>PDB  HSSP</td>\n",
       "      <td>2ci2  Barnase</td>\n",
       "      <td>Barnase and chymotrypsin inhibitor</td>\n",
       "      <td>NN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90</td>\n",
       "      <td>Supervised Learning: Regression</td>\n",
       "      <td>Drug Design</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>Ladunga I., \\n1991, \\nHungary</td>\n",
       "      <td>UniProt</td>\n",
       "      <td>Signal peptide</td>\n",
       "      <td>Novel predicted signal peptides</td>\n",
       "      <td>NN (Tiling algorithm)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "      <td>Supervised Learning: Classification</td>\n",
       "      <td>Protein site prediction</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     1  \\\n",
       "0           First Author/Year of  Publication/ Country   \n",
       "1                               Burak T., 2021, Turkey   \n",
       "2                               Song J., 2021, \\nChina   \n",
       "3                                 Verma N., 2021, \\nUS   \n",
       "4                 Li H., 2020, \\nFrance, \\nHong Kong,    \n",
       "5                               Kwon Y., \\n2020, Korea   \n",
       "6                                   Wang M.., 2020, US   \n",
       "7                                   Qin Z., \\n2020, US   \n",
       "8                          Zheng W., 2020, \\nChina, US   \n",
       "9                            Yu C., 2020, \\nTaiwan, US   \n",
       "10                              Yang H., 2020, \\nChina   \n",
       "11                            Wang X., \\n2020, \\nChina   \n",
       "12                             Maia E., 2020, \\nBrazil   \n",
       "13                               Shroff R., 2020, \\nUS   \n",
       "14                 Mahmoud A., \\n2020, Switzerland, US   \n",
       "15       Wardah W., 2020, \\nAustralia, Fiji, Japan, US   \n",
       "16                           Liang M., \\n2020, \\nChina   \n",
       "17             Feger G., 2020,\\nCzech Republic, France   \n",
       "18                                 Dai W., China, 2020   \n",
       "19                                 Bond P., 2020, \\nUK   \n",
       "20                     Nie J., 2020, Singapore, Taiwan   \n",
       "21                         Savojardo C., 2020, \\nItaly   \n",
       "22                    Du Z., 2020, \\nChina, Russia, US   \n",
       "23                                Li C., 2019, \\nChina   \n",
       "24                                 Lin J., 2019, China   \n",
       "25                                   Fang C., 2019, US   \n",
       "26                                  Wan C., 2019, \\nUK   \n",
       "27                               Luo X., 2019, \\nChina   \n",
       "28                        Mirabello C., 2019, \\nSweden   \n",
       "29                         Savojardo C., 2019, \\nItaly   \n",
       "30                    Rifaioglu A., 2019, \\nTurkey, UK   \n",
       "31                       Han X., \\n2019, Singapore, US   \n",
       "32                              Zhao Z., 2019, \\nChina   \n",
       "33                              Long S., 2019, \\nChina   \n",
       "34                               Zhang Y., China, 2019   \n",
       "35                          Kaleel M., 2019, \\nIreland   \n",
       "36   Schantz M., 2019, \\nArgentina, \\nDenmark, \\nMa...   \n",
       "37                             Luo L., \\n2019, \\nChina   \n",
       "38                    Haberal I., 2019, Norway, Turkey   \n",
       "39                      Heinzinger M., 2019, \\nGermany   \n",
       "40                                Zhang D., 2019, \\nUS   \n",
       "41                       Fang C., 2019, \\nChina, Japan   \n",
       "42                           Cui Y., \\n2019, \\nChina\\n   \n",
       "43                  Taherzadeh G., 2019, Australia, US   \n",
       "44                                    Guo Y., 2019, US   \n",
       "45                              Yang R., \\n2019, China   \n",
       "46                                  Zheng W., 2019, US   \n",
       "47                                Wang D., 2019, China   \n",
       "48                               Fu H.,\\n2019, \\nChina   \n",
       "49                          Pagès G., \\n2019, \\nFrance   \n",
       "50                            Torng  W., \\n2019,  \\nUS   \n",
       "51                                 Yu C., \\n2019, \\nUS   \n",
       "52                                  Huang L., 2018, US   \n",
       "53                 Hanson J., Australia, China, \\n2018   \n",
       "54                  Hanson J., 2018, Australia, China    \n",
       "55                              Zhao X., 2018, \\nChina   \n",
       "56            O’Connell J., 2018, Australia, China, US   \n",
       "57                         Xue L., \\n2018, \\nChina, US   \n",
       "58                    Feinberg E., \\n2018, \\nChina, US   \n",
       "59                            Zhang L.,\\n2018, \\nChina   \n",
       "60                             Zhang B., 2018, \\nChina   \n",
       "61                               Luo F., 2018, \\nChina   \n",
       "62                       Long H.,  \\n2018, \\nChina, US   \n",
       "63                            Frasca M., 2018, \\nItaly   \n",
       "64                              Hu B., \\n2018, \\nChina   \n",
       "65                             Le N., \\n2018, \\nTaiwan   \n",
       "66                              Sunseri J., 2018, \\nUS   \n",
       "67                              Cuperus J., 2018, \\nUS   \n",
       "68                              Li H., \\n2018, \\nChina   \n",
       "69                       Popova M., 2018, \\nRussia, US   \n",
       "70                             Zafeiris D., 2018, \\nUK   \n",
       "71                               Zhao B., \\n2018, \\nUS   \n",
       "72                            Hochuli J., \\n2018, \\nUS   \n",
       "73                         Khurana S., 2018, Qatar, US   \n",
       "74              Makrodimitris S., \\n2018,  Netherlands   \n",
       "75                                 Fang C., 2018, \\nUS   \n",
       "76                              Adhikari B., 2017,\\nUS   \n",
       "77                      Armenteros J., 2017, \\nDenmark   \n",
       "78                         Jimenez J., \\n2017, \\nSpain   \n",
       "79                               Wang S., \\n2017, \\nUS   \n",
       "80                               Vang Y., \\n2017, \\nUS   \n",
       "81                        Müller A., 2017, Switzerland   \n",
       "82                          Szalkai B., 2017,\\nHungary   \n",
       "83                       Szalkai B., \\n2017, \\nHungary   \n",
       "84                                 Cao R, \\n2017, \\nUS   \n",
       "85                                Ragoza M., 2017,\\nUS   \n",
       "86                            Yeh C., \\n2017, \\nUK, US   \n",
       "87                      Al-Gharabli S., 2015, \\nJordan   \n",
       "88                        Yang J., \\n2015, \\nChina, US   \n",
       "89               Simha R., 2015, \\nCanada, Germany, US   \n",
       "90                         Khan Z., \\n2014, \\nPakistan   \n",
       "91                                 Li Y., \\n2014, \\nUS   \n",
       "92                               Murphy G., 2014, \\nUS   \n",
       "93                     Folkman L., \\n2014, \\nAustralia   \n",
       "94                                 Li Z., \\n2014, \\nUS   \n",
       "95                       Volpato V., \\n2013, \\nIreland   \n",
       "96         Wong K., \\n2013, \\nCanada, US, Saudi Arabia   \n",
       "97                           Traoré S., 2013, \\nFrance   \n",
       "98                                 Qi Y., \\n2012, \\nUS   \n",
       "99                     Eisenbeis S., \\n2012, \\nGermany   \n",
       "100                             Daniels N., 2012, \\nUS   \n",
       "101                            Durrant J.,  2011, \\nUS   \n",
       "102                         Ebina T.,  \\n2011, \\nJapan   \n",
       "103                     Ashkenazy H., \\n2011, \\nIsrael   \n",
       "104                               Yang Y.,\\n2011, \\nUS   \n",
       "105                         Latek D., \\n2010, \\nPoland   \n",
       "106                              Zhao F., \\n2010, \\nUS   \n",
       "107                Briesemeister S., \\n2010, \\nGermany   \n",
       "108                               Lin G., \\n2010, \\nUS   \n",
       "109                       Tian J., \\n2010, \\nChina, US   \n",
       "110                               Bostan B., 2009,\\nUS   \n",
       "111                               Hong E.,\\n2009, \\nUS   \n",
       "112                          Özen A., \\n2009, \\nTurkey   \n",
       "113                  Ebrahimpour A., \\n2008,  Malaysia   \n",
       "114                            Zhu X., \\n2008, \\nChina   \n",
       "115                         Huang W., \\n2008, \\nTaiwan   \n",
       "116                                 Wu S.,\\n2008, \\nUS   \n",
       "117                             Katzman S., 2008, \\nUS   \n",
       "118                              Shah R., \\n2008, \\nUS   \n",
       "119                         Armstrong K., \\n2008, \\nUS   \n",
       "120                         Shamim M., \\n2007, \\nIndia   \n",
       "121                              Liao J., \\n2007, \\nUS   \n",
       "122                         Raveh B., \\n2007, \\nIsrael   \n",
       "123                   Zimmermann O., \\n2006, \\nGermany   \n",
       "124                             Sidhu A., \\n2006, \\nUK   \n",
       "125                          Hung C., \\n2006, \\nTaiwan   \n",
       "126                        Capriotti E., 2005, \\nItaly   \n",
       "127                                 Hu C.,\\n2004, \\nUS   \n",
       "128                          Gutteridge A., 2003, \\nUK   \n",
       "129                     Bindslev C., \\n2002, \\nDenmark   \n",
       "130                              Russ W., \\n2002, \\nUS   \n",
       "131                          Rossi A., \\n2001, \\nItaly   \n",
       "132                      Ladunga I., \\n1991, \\nHungary   \n",
       "\n",
       "                                                     2  \\\n",
       "0                                             Database   \n",
       "1                                              UniProt   \n",
       "2                                      PDB\\nSwiss-Prot   \n",
       "3                               DrugBank\\nMatador\\nPDB   \n",
       "4      PDB,\\nPubChem,\\nZINC,\\nChEMB,L\\nBindingDB,\\nHTS   \n",
       "5                                         PDBind-2016    \n",
       "6      \\nSKEMPI 1.0, 2.0 dataSet\\nAB-Bind S645 dataset   \n",
       "7                                                  PDB   \n",
       "8                                            SCOPe2.07   \n",
       "9                                                  PDB   \n",
       "10                                          SCOPe 2.07   \n",
       "11                                 KIBA, Davis dataset   \n",
       "12                                  Multiple databases   \n",
       "13                                                 PDB   \n",
       "14                                                 PDB   \n",
       "15                                              BioLiP   \n",
       "16                                                 PDB   \n",
       "17                                                 PDB   \n",
       "18                          Reactome DB & InBio Map DB   \n",
       "19                                                 PDB   \n",
       "20                                             UniProt   \n",
       "21                         UniProtKB\\nGOA,\\nDeepMitoDB   \n",
       "22                                    CAFA3, SwissProt   \n",
       "23                                LE dataset from SCOP   \n",
       "24                                            DrugBank   \n",
       "25                                             UniProt   \n",
       "26                                UniProtKB/Swiss-Prot   \n",
       "27                                            Transfac   \n",
       "28                                                 PDB   \n",
       "29                                UniprotKB/Swiss-Prot   \n",
       "30                                UniProtKB/Swiss-Prot   \n",
       "31                                                eSol   \n",
       "32                     Monomers and dimers from author   \n",
       "33   Jpred dataset\\ncullpdb dataset\\nUniRef90\\nUniProt   \n",
       "34                                                 PDB   \n",
       "35                                                 PDB   \n",
       "36                                         PDB, PISCES   \n",
       "37   BioCreative II,\\nBioCreative III,\\nBioCreative...   \n",
       "38                                                 PDB   \n",
       "39                                        UniProt, PDB   \n",
       "40                                 DIP, HPRD,\\nUniProt   \n",
       "41                                                 PDB   \n",
       "42                                              BioLip   \n",
       "43      Uniprot, dbPTM, Uniprep, UnicarKB, GlycoProtDB   \n",
       "44                               CB513, CASP10, CASP11   \n",
       "45                                Datasets A(CPLM),B,C   \n",
       "46                                              CASP13   \n",
       "47                                          CASP11, 12   \n",
       "48                                                PLMD   \n",
       "49                                                CASP   \n",
       "50                             PROSITE     NOS dataset   \n",
       "51                                                 PDB   \n",
       "52                                           DIP, HPRD   \n",
       "53                                 PISCES\\nCASP12\\nPDB   \n",
       "54                                        PDB\\nUniProt   \n",
       "55                                                PLMD   \n",
       "56                                        SPIN dataset   \n",
       "57                                  Swiss-Prot, TrEMBL   \n",
       "58                                       PDB Bind 2007   \n",
       "59                                      UniProt, \\nDIP   \n",
       "60                         PDB, PISCES, TR5534 Dataset   \n",
       "61   Phospho.ELM, PhosphositePlus, HPRD, dbPTM, SysPTM   \n",
       "62                                             UniProt   \n",
       "63                                         STRING   GO   \n",
       "64   DDI    SIDER  TWOSIDES   HPRD   Drug Bank  Off...   \n",
       "65                                         UniProt  GO   \n",
       "66            D3R Grand challenge 3  Grand challenge 3   \n",
       "67       5′ UTR library of 50-nt-long random sequences   \n",
       "68                                  HPRD  DIP   HIPPIE   \n",
       "69                               PHYSPROP  ChEMBL  KKB   \n",
       "70                               GEO  Array Expression   \n",
       "71                                       DisProt   PDB   \n",
       "72                                                 PDB   \n",
       "73                                     pepcDB database   \n",
       "74                       Arabidopsis thaliana proteins   \n",
       "75                             PDB\\nJPRED\\nCASP\\nCB513   \n",
       "76                                     DNCON \\nDataset   \n",
       "77                                             UniProt   \n",
       "78                                               scPDB   \n",
       "79                                 Pfam   CASP   CAMEO   \n",
       "80                               IEDB  MHCBN SYFPEITHI   \n",
       "81                                     ADAM  APD  DADP   \n",
       "82                           Swiss-Prot, UniProt, \\nGO   \n",
       "83                                         UniProt  GO   \n",
       "84                                             UniProt   \n",
       "85                                         PDB  ChEMBL   \n",
       "86   JSON database:  centers of mass and geometric ...   \n",
       "87                                                 PDB   \n",
       "88                    PDB  SPx dataset  PDBCYS dataset   \n",
       "89                                      DBMLoc dataset   \n",
       "90                                              BRENDA   \n",
       "91                                                 PDB   \n",
       "92                                     DND_4HB protein   \n",
       "93                                            ProTherm   \n",
       "94                                              PISCES   \n",
       "95                                      ENZYME UniProt   \n",
       "96                 Protein Binding Microarray data set   \n",
       "97                                                 PDB   \n",
       "98                     Standard benchmark, CB513  DSSP   \n",
       "99                                                 PDB   \n",
       "100                                               SCOP   \n",
       "101                                          PDB  MOAD   \n",
       "102                                    DS-All  dataset   \n",
       "103                                                PDB   \n",
       "104                                              SPINE   \n",
       "105  10 globular proteins 216 residues & S100A1 pro...   \n",
       "106                                               CASP   \n",
       "107                                            UniProt   \n",
       "108                                                PDB   \n",
       "109                                      ProTherm  PDB   \n",
       "110                                               KEGG   \n",
       "111  Standard rotamer library  Expanded rotamer lib...   \n",
       "112                                           ProTherm   \n",
       "113                                            GenBank   \n",
       "114                                                PDB   \n",
       "115                                        UniProt  GO   \n",
       "116                                                PDB   \n",
       "117                                        PDB  PISCES   \n",
       "118                                          D Dataset   \n",
       "119                                                PDB   \n",
       "120                      D-B dataset  Ext. D-B dataset   \n",
       "121                                            GenBank   \n",
       "122                                                PDB   \n",
       "123                                                PDB   \n",
       "124                                         Swiss-Prot   \n",
       "125                                               NCBI   \n",
       "126                                           ProTherm   \n",
       "127                            WhatIF database UniProt   \n",
       "128                                                PDB   \n",
       "129  20 Patients with allergy to Macrozoarces ameri...   \n",
       "130                                                NaN   \n",
       "131                                          PDB  HSSP   \n",
       "132                                            UniProt   \n",
       "\n",
       "                                                     3  \\\n",
       "0                                Initial Scaffold (ID)   \n",
       "1          Protein sequence from 60 different families   \n",
       "2                                 ATP-binding proteins   \n",
       "3                                  Human, \\nC. Elegans   \n",
       "4    Chemical\\nEstrogen receptor α (ERα)\\nAnaplasti...   \n",
       "5        VEGFR2 kinase domain And adenosine deaminase    \n",
       "6                            Protein-protein complexes   \n",
       "7    Phi–psi angle and sequence of natural protein,...   \n",
       "8                                                  NaN   \n",
       "9                                α-helix rich proteins   \n",
       "10                                                 NaN   \n",
       "11                               Kinase protein family   \n",
       "12           structure-based virtual screening (SBVS)    \n",
       "13                                                 NaN   \n",
       "14             HIV-1 protease, dihydrofolate reductase   \n",
       "15   Positive (binding) or negative (non-binding), ...   \n",
       "16      Relative angle of (C − Cα − C) principal plane   \n",
       "17                      Peptide amphiphiles\\nscaffolds   \n",
       "18                                Human essential gene   \n",
       "19   Only residues\\nwith side chains longer than be...   \n",
       "20   acetyl-lysine’ (S1), ‘crotonyl-lysine’ (S2),\\n...   \n",
       "21   Human, mouse, fly, yeast and Arabidopsis thaliana   \n",
       "22                                 Human, \\nC. Elegans   \n",
       "23                              Multiple superfamilies   \n",
       "24      druggable proteins  and non-druggable proteins   \n",
       "25   Proteins from datasets BT426 and BT6376, conta...   \n",
       "26                                      Human proteins   \n",
       "27                                       DNA sequences   \n",
       "28                                                 NaN   \n",
       "29                             mitochondrial proteins    \n",
       "30                                                 NaN   \n",
       "31          ell-free protein expression from E.\\ncoli    \n",
       "32                     Monomers and dimers from author   \n",
       "33                              Multiple superfamilies   \n",
       "34                     PDNA-543, PDNA-224 and PDNA-316   \n",
       "35   amino acids are categorised into four classes ...   \n",
       "36                                  Crystal structures   \n",
       "37                                PPI protein articles   \n",
       "38   Metal-binding of histidines & Cysteines amino ...   \n",
       "39                                TS115\\nCB513\\nCASP12   \n",
       "40   D. melanogaster, S. cerevisiae,\\nE. coli, C. e...   \n",
       "41             MoRF-containing membrane protein chains   \n",
       "42                           Fourteen binding residues   \n",
       "43                                       glycoprotein    \n",
       "44                                   Protein sequences   \n",
       "45                        Proteins and reducing sugars   \n",
       "46                             Query sequence profiles   \n",
       "47                                          Caspase 14   \n",
       "48   Sequences and physicochemical properties of pr...   \n",
       "49                                            Model QA   \n",
       "50                      Protein structure as 3D images   \n",
       "51   194l (lysozyme), 107m (myoglobin), 6cgz (β-bar...   \n",
       "52                                   PPI network graph   \n",
       "53                                5N5EA \\n6FI2A\\n6FQ3A   \n",
       "54   Primary amino-acid-sequence, proteins from CASP12   \n",
       "55                                              Lysine   \n",
       "56                                                 NaN   \n",
       "57                                   Secretory protein   \n",
       "58   Scaffold split for grouping ligands in common ...   \n",
       "59          S. cerevisiae, H. sapiens, and M. musculus   \n",
       "60                               CASP10, 11, 12 and 13   \n",
       "61                               Kinase protein family   \n",
       "62              PseAAC  Hydroxyproline & hydroxylysine   \n",
       "63   Organisms:   Homo sapiens (human)  S.cerevisia...   \n",
       "64                            Semantic meta-paths  ADR   \n",
       "65   Rab GGT activity  Rab GDI activity  Rab GTPase...   \n",
       "66            Input ligand  SMILES protein FASTA  CSAR   \n",
       "67                      Yeast Saccharomyces cerevisiae   \n",
       "68   Primary sequence  Escherichia coli,  Drosophil...   \n",
       "69                                        SMILE string   \n",
       "70   Amyloid beta precursor protein, microtubule as...   \n",
       "71   Intrinsically disordered proteins (IDPs), intr...   \n",
       "72                         Ligands SMILE Protein FASTA   \n",
       "73   k-mer structure and additional sequence and st...   \n",
       "74                        Arabidopsis thaliana protein   \n",
       "75           Different super-families,\\nCASP10, 11, 12   \n",
       "76                                                 NaN   \n",
       "77             Protein sequence,  Sequence information   \n",
       "78                             PDB ID File or PDB file   \n",
       "79   150 Pfam families 105 CASP11 test proteins  76...   \n",
       "80               Human leukocyte antigen (HLA) complex   \n",
       "81        Antimicrobial peptide   Amino acid sequences   \n",
       "82   Thyroid hormone,  phenol-containing\\ncompound,...   \n",
       "83                                         Classes.tre   \n",
       "84                                    Protein sequence   \n",
       "85   Spatial and chemical  features of protein-liga...   \n",
       "86   Helical repeat proteins, Center of mass (CoM) ...   \n",
       "87                  Amino acid sequence hydrophobicity   \n",
       "88                                 Amino acid sequence   \n",
       "89                                                 NaN   \n",
       "90            Amino Acid sequence  and alkaline enzyme   \n",
       "91                                             E. coli   \n",
       "92                                     DND_4HB protein   \n",
       "93        Protein sequence and amino acid substitution   \n",
       "94       TL2282 dataset  TS500 dataset  TR1532 dataset   \n",
       "95   Oxidoreductase, transferase, hydrolase, lyase,...   \n",
       "96                                        DNA sequence   \n",
       "97                                3D protein structure   \n",
       "98                     PSI-BLAST  amino acid embedding   \n",
       "99   (βα)8-barrel and the flavodoxin-like fold,CheY...   \n",
       "100  Protein sequence, 207 beta structural SCOP sup...   \n",
       "101                             Crystal structure data   \n",
       "102                                   Protein sequence   \n",
       "103                               3D protein structure   \n",
       "104                                   Protein sequence   \n",
       "105              10 globular proteins & S100A1 protein   \n",
       "106  ( PSSM) Position-specific scoring matrix  gene...   \n",
       "107                                   Protein sequence   \n",
       "108  Protein Folding Rates.  Predicting protein fol...   \n",
       "109                                      3D structures   \n",
       "110                           Given a species proteome   \n",
       "111  fn3: Derived from protein Fn3, tenth human fib...   \n",
       "112  Structure-based features: amino acid substitut...   \n",
       "113                             Geobacillus sp. strain   \n",
       "114                              223 scaffold proteins   \n",
       "115  SCL12, SCL16  Sequence-based, GO terms, protei...   \n",
       "116                               PDB protein sequence   \n",
       "117  Amino acid sequence of a protein of unknown st...   \n",
       "118                                   Protein sequence   \n",
       "119                                   Protein sequence   \n",
       "120  Structural information of amino acid residue a...   \n",
       "121  Proteinase K-catalyzed hydrolysis of the tetra...   \n",
       "122  TIM-barrel fold 1YPI.  Whole β-sheets global s...   \n",
       "123                                   Protein sequence   \n",
       "124  signal peptides and non-secretory proteins fro...   \n",
       "125        Nucleocapsid (nsp1) of a coronavirus family   \n",
       "126                         Protein tertiary structure   \n",
       "127  3D coarse-grained structure from protein seque...   \n",
       "128  Amino acid sequence of quinolate phosphoribosy...   \n",
       "129                            Macrozoarces americanus   \n",
       "130  SH3 domain  GroEL minichaperone  WW domain pro...   \n",
       "131                                      2ci2  Barnase   \n",
       "132                                     Signal peptide   \n",
       "\n",
       "                                                     4  \\\n",
       "0                                     Designed Protein   \n",
       "1         Protein family classification/identification   \n",
       "2                         Protein–ATP Binding Residues   \n",
       "3                          Protein ligand interactions   \n",
       "4                               Protein-ligand complex   \n",
       "5    Prediction of affinity-binding of a protein-li...   \n",
       "6         Protein-ligand binding affinity predictions    \n",
       "7       Protein design of fold alpha-helical structure   \n",
       "8                            Protein domain boundaries   \n",
       "9                               De novo protein design   \n",
       "10                      Contact map protein prediction   \n",
       "11               Predicte Drug-target binding affinity   \n",
       "12                                    Drug development   \n",
       "13              amino acids association guide mutation   \n",
       "14   Hydration\\nsite occupancy and thermodynamics p...   \n",
       "15            Prediction Protein-peptide binding sites   \n",
       "16                       Enzymatic function prediction   \n",
       "17                Amphiphilic peptide scaffolds design   \n",
       "18                       Predict human essential genes   \n",
       "19      Prediction the correctness of protein residues   \n",
       "20                  Identification of Lysine PTM sites   \n",
       "21              Protein sub-mitochondrial localization   \n",
       "22                       Automated function prediction   \n",
       "23   Detect the structural\\nmotifs related with the...   \n",
       "24                              Drug target prediction   \n",
       "25                               Beta-turn prediction    \n",
       "26                                 Function prediction   \n",
       "27                      predicting DNA–protein binding   \n",
       "28                                   Method prediction   \n",
       "29           sub-mitochondrial cellular\\nlocalization.   \n",
       "30                                 GO terms prediction   \n",
       "31                                  Protein solubility   \n",
       "32                         Protein-protein interaction   \n",
       "33              Protein secondary structure prediction   \n",
       "34          Identification of DNA–protein binding site   \n",
       "35        Prediction of relative solvent accessibility   \n",
       "36           Prediction of protein structural features   \n",
       "37                         Protein-protein interaction   \n",
       "38            Prediction of metal-binding in proteins    \n",
       "39                     Protein sequence representation   \n",
       "40   Protein-protein interactions and protein famil...   \n",
       "41     Molecular recognition features MoRFs prediction   \n",
       "42           Protein-ligand binding residue prediction   \n",
       "43                     N- and O-linked\\nglycosylation    \n",
       "44                         Protein secondary structure   \n",
       "45                        Glycation product prediction   \n",
       "46             Automated structure prediction pipeline   \n",
       "47                        Protein structure refinement   \n",
       "48   Predict Lysine ubiquitination sites in large-s...   \n",
       "49                    Protein model quality assessment   \n",
       "50                   Protein functional site detection   \n",
       "51   Generation of  audible sound  from amino acids...   \n",
       "52                         Protein-Protein interaction   \n",
       "53   Sequence-based prediction of one dimensional s...   \n",
       "54                      Protein contact map prediction   \n",
       "55                            Lysine acetylation sites   \n",
       "56                        Sequences profile compatible   \n",
       "57            Protein sequence into T3SEs or non-T3SEs   \n",
       "58                       Molecular Property Prediction   \n",
       "59             Predicting Protein-Protein interactions   \n",
       "60                Prediction of performance of protein   \n",
       "61                             Protein phosphorylation   \n",
       "62                      Predicting hydroxylation sites   \n",
       "63         AFP (Automated Protein Function Prediction)   \n",
       "64                     meta-path-based proximities ADR   \n",
       "65                      Classify Rab protein molecules   \n",
       "66                   Cathepsin S  model ligand protein   \n",
       "67                          Predict protein expression   \n",
       "68                  Prediction of protein interactions   \n",
       "69                        Drug design (de novo design)   \n",
       "70         Biomarker discovery for Alzheimer's disease   \n",
       "71                                                 NaN   \n",
       "72                     Identify protein ligand scoring   \n",
       "73                               Solubility prediction   \n",
       "74                         Protein function prediction   \n",
       "75              Protein secondary structure prediction   \n",
       "76                      Contact map protein prediction   \n",
       "77           Predict protein sub-cellular localization   \n",
       "78    Predict protein-ligand-binding sites Drug design   \n",
       "79                                               5f5pH   \n",
       "80              HLA class I-peptide binding prediction   \n",
       "81   Design of new peptides combinatorial de novo p...   \n",
       "82      protein classification by amino-acid\\nsequence   \n",
       "83      Hierarchical Biological Sequence Classiﬁcation   \n",
       "84                         Protein function prediction   \n",
       "85             Protein-ligand score for drug discovery   \n",
       "86             Designed helical repeat proteins (DHRs)   \n",
       "87   Prediction of dihedral angles physiochemical p...   \n",
       "88     Structure  prediction of cysteine-rich proteins   \n",
       "89                   Protein multi-location prediction   \n",
       "90                                    Enzyme catalysis   \n",
       "91   Designs of improved enzymes and enzymes with n...   \n",
       "92                Design an up-down four-helix bundle.   \n",
       "93      Model designed for a specific type of mutation   \n",
       "94                         Sequence profile prediction   \n",
       "95   Acid-residue frequency derived from multiple s...   \n",
       "96                                 DNA-motif discovery   \n",
       "97   Structure-based computational  protein design ...   \n",
       "98          Prediction of local properties in proteins   \n",
       "99                                       Enzyme design   \n",
       "100  Detection for beta-structural proteins into th...   \n",
       "101            Identiﬁcation of small-molecule ligands   \n",
       "102                                   Domain predictor   \n",
       "103                             Contact map prediction   \n",
       "104                     Single-method fold recognition   \n",
       "105  Predicted Nuclear Overhauser Effect signals on...   \n",
       "106                                    Protein folding   \n",
       "107           Predict protein subcellular localization   \n",
       "108  Protein folding kinetic rate and real-value fo...   \n",
       "109  Effect on single or multi-site mutation on pro...   \n",
       "110               Predict homologous signaling pathway   \n",
       "111  Tenth human fibronectin, D44.1 and DI.3 antibo...   \n",
       "112                Single-site amino acid substitution   \n",
       "113  Lipase production Syncephalastrum racemosum, P...   \n",
       "114  Pocket residues of ribose-binding protein (2dr...   \n",
       "115  Prediction method for predicting subcellular l...   \n",
       "116                          Protein-contact predictor   \n",
       "117                         Local structure prediction   \n",
       "118                                 Homology detection   \n",
       "119    Protein engineering space of foldable sequences   \n",
       "120                            Protein fold prediction   \n",
       "121                              Proteinase K variants   \n",
       "122  Existence of α-helices, parallel β-sheets, ant...   \n",
       "123                     Prediction of dihedral regions   \n",
       "124                             Predict signal peptide   \n",
       "125                          Predict protein functions   \n",
       "126                       Protein stability prediction   \n",
       "127                         Optimal non-linear scoring   \n",
       "128                                Predict active site   \n",
       "129  Investigate potential allergenicity of Ice Str...   \n",
       "130   Thermostable consensus phytase, 84.5 kDa protein   \n",
       "131                 Barnase and chymotrypsin inhibitor   \n",
       "132                    Novel predicted signal peptides   \n",
       "\n",
       "                                                     5  \\\n",
       "0                                             ML Model   \n",
       "1    FIBHASH (based on Fibonacci numbers), RNN, LST...   \n",
       "2                                        CNN\\nLightGBM   \n",
       "3                                                  DNN   \n",
       "4    RF, BRT, kNN, NN, SVM,  GBDT,\\nmulti-task\\nDNN...   \n",
       "5                                               3D CNN   \n",
       "6          site-specific persistent homology, CNN, GBT   \n",
       "7    Multi-scale Neighborhood-based Neural Network ...   \n",
       "8                                               FUpred   \n",
       "9                                            RNN, LSTM   \n",
       "10                                                 GAN   \n",
       "11                                            CNN\\nGCN   \n",
       "12        ANN;\\nSVM\\nBT\\nDT\\nkNN\\nEnsemble methods\\nGA   \n",
       "13                                              3D CNN   \n",
       "14                                                 CNN   \n",
       "15                                                 CNN   \n",
       "16                                           RNN, LSTM   \n",
       "17                                             SVM, RF   \n",
       "18                                                 SVM   \n",
       "19                                             NN, MLP   \n",
       "20   RF,\\n SVM, MNB,\\nLR, Max Entropy, KNN,\\nCNN,\\nMLP   \n",
       "21                                  DeepMito, \\n1D-CNN   \n",
       "22                                            NLP, CNN   \n",
       "23                   MotifCNN and MotifDCNN\\nSVM \\nCNN   \n",
       "24                                             SVM, GA   \n",
       "25     HMM, CNN, deep dense inception network (DeepDIN   \n",
       "26              Deep mahout neural network (DMNN), SVM   \n",
       "27                                                 CNN   \n",
       "28                                                 DNN   \n",
       "29                                                 CNN   \n",
       "30                                                 DNN   \n",
       "31                                            DNN\\nGAN   \n",
       "32                                                LSTM   \n",
       "33                                                 CNN   \n",
       "34   Adaptive Synthetic Sampling (ADASYN) approach\\...   \n",
       "35                                           BRNN, CNN   \n",
       "36                                           CNN, LSTM   \n",
       "37   Knowledge-enriched Self-Attention Convolutiona...   \n",
       "38                                      CNN, LSTM, RNN   \n",
       "39                                ELMo\\nCharCNN  LSTMs   \n",
       "40                                      Multimodal DNN   \n",
       "41                                                DCNN   \n",
       "42                                                DCNN   \n",
       "43                                             DNN SVM   \n",
       "44                                         ACNN, BLSTM   \n",
       "45                                            RNN, CNN   \n",
       "46                     ZhangServer and QUARK pipelines   \n",
       "47   Monte Carlo, multi-objective particle swarm op...   \n",
       "48                                   CNN  DL   DeepUbi   \n",
       "49                                              3D CNN   \n",
       "50                                     DL  3D CNN  SVM   \n",
       "51                                           RNN  LSTM   \n",
       "52   Evolution neural network based regularized Lap...   \n",
       "53                                     CNN, 2D-BRLSTM    \n",
       "54       CNN, 2D bidirectional recurrent LSTM (BRLSTM)   \n",
       "55                                            CNN\\nDNN   \n",
       "56                                                 DNN   \n",
       "57                                                 CNN   \n",
       "58                                     RNN DTNN & MPNN   \n",
       "59                                 DNN, Adam Algorithm   \n",
       "60                                      CNN, RNN, BRNN   \n",
       "61                                                 CNN   \n",
       "62                                 HDL  CNN  LSTM  RNN   \n",
       "63                              COSNet  ParCOSNet  HNN   \n",
       "64                                              SDHINE   \n",
       "65                                                 CNN   \n",
       "66                                                 CNN   \n",
       "67                                                 CNN   \n",
       "68                                      DNN  CNN  LSTM   \n",
       "69                           Stack-RNN  LSTM.  ReLeaSE   \n",
       "70                                                 ANN   \n",
       "71                                             ANN  DT   \n",
       "72                                                 CNN   \n",
       "73                                       CNN  DL  FFNN   \n",
       "74                                           KNN  LSDR   \n",
       "75   Deep\\ninception-inside-inception (Deep3I) network   \n",
       "76                                                 CNN   \n",
       "77             CNN  RNN  BLSTM  FFNN  Attention models   \n",
       "78                                             3D DCNN   \n",
       "79                                         UDNN  RNN 2   \n",
       "80                                                 CNN   \n",
       "81                                           RNN  LSTM   \n",
       "82                                                 DNN   \n",
       "83                                            ANN  DNN   \n",
       "84                                                 RNN   \n",
       "85                                            CNN  SGD   \n",
       "86                        GA  multithreaded processing   \n",
       "87                                                 ANN   \n",
       "88                                            HMM  SVR   \n",
       "89                                           MDLoc  BN   \n",
       "90                              KNN  SVM  MLP  PNN  DT   \n",
       "91                                                 NaN   \n",
       "92                               Computational folding   \n",
       "93                                  SFFS  SVM  EASE-MM   \n",
       "94                                            SPIN  NN   \n",
       "95                              N-to-1 Neural Networks   \n",
       "96                                            Kmer-HMM   \n",
       "97                                                 CFN   \n",
       "98                                                 DNN   \n",
       "99                                                 NaN   \n",
       "100                                           HMM  MRF   \n",
       "101                           ANN scoring function map   \n",
       "102                                      DROP  SVM  RF   \n",
       "103                                                WMC   \n",
       "104                                SPARKS-X  Algorithm   \n",
       "105                                           CABS  MC   \n",
       "106                                           CNF  SVM   \n",
       "107                                                 NB   \n",
       "108                                           SVM  SVR   \n",
       "109                                       RFR  SVM  RF   \n",
       "110                                                PSP   \n",
       "111                                             BroMAP   \n",
       "112                                  SVM  KNN  DT  SVR   \n",
       "113              ANN  FFNN  IBP  BBP    QP     GA   LM   \n",
       "114                                                DGA   \n",
       "115                                       GA  SVM  KNN   \n",
       "116                                        MUSTER  SVM   \n",
       "117                                     Multi-layer NN   \n",
       "118                                                SVM   \n",
       "119                                                 GR   \n",
       "120                                                SVM   \n",
       "121     RR   Lasso PLSR SVMR LPSVMR  LPBoosR  MR  ORMR   \n",
       "122                                 K-means Clustering   \n",
       "123                SVM  C-SVM algorithm implementation   \n",
       "124                                      BBFNN  NN  DT   \n",
       "125                                 DFS  HMM  GA  AGCT   \n",
       "126                                                SVM   \n",
       "127           SVM non-linear Gaussian kernel functions   \n",
       "128                                           NN  FFNN   \n",
       "129                                                 DT   \n",
       "130                                                NaN   \n",
       "131                                                 NN   \n",
       "132                              NN (Tiling algorithm)   \n",
       "\n",
       "                                                     6  \\\n",
       "0                                     Software / Sever   \n",
       "1                                                  NaN   \n",
       "2    TensorFlow, Keras\\nhttps://github.com/tlsjz/AT...   \n",
       "3             GitHub (https://github.com/ekraka/SSnet)   \n",
       "4    Descriptor data bank\\nODDT\\nBINANA\\nRF-Score-v...   \n",
       "5                                    Keras, Tensorflow   \n",
       "6    TopNetTree, Keras\\nhttps://doi.org/10.24433/CO...   \n",
       "7             Tensorflow\\nhttps://github.com/IBM/mnnn.   \n",
       "8          https://zhanglab.ccmb.med.umich.edu/FUpred/   \n",
       "9    TensorFlow, \\nhttps://github.com/tensorflow/ma...   \n",
       "10   Keras, Tensorflow\\nhttps://github.com/melissay...   \n",
       "11                                                 NaN   \n",
       "12                                                 NaN   \n",
       "13                        Theano\\nwww.\\nmutcompute.com   \n",
       "14         https://hub.docker.com/r/lilllab/\\nwatsite3   \n",
       "15      PyTorch, https://github.com/WafaaWardah/Visual   \n",
       "16                                          TensorFlow   \n",
       "17                                              SasFit   \n",
       "18                                                 NaN   \n",
       "19                                                CCP4   \n",
       "20   Tensorflo, https://github.com/khanhlee/\\nlysin...   \n",
       "21                                                 NaN   \n",
       "22                                   Keras, TensorFlow   \n",
       "23                                          TensorFlow   \n",
       "24      https://github.com/QUST-AIBBDRC/GA-Bagging-SVM   \n",
       "25   Tensorflow, Keras \\nhttp://dslsrv8.cs.missouri...   \n",
       "26        Keras,\\nhttps://github.com/psipred/STRING2GO   \n",
       "27   Keras, Tensorflow\\nhttps://github.com/gao-lab/...   \n",
       "28   Keras, TensorFlow\\nhttps://bitbucket.org/clami...   \n",
       "29              http://busca.biocomp.unibo.it/deepmito   \n",
       "30      Tensorflow,\\nhttps://github.com/cansyl/DEEPred   \n",
       "31                                                 NaN   \n",
       "32                                                 NaN   \n",
       "33                                     TensorFlow\\nN/A   \n",
       "34                                              Theano   \n",
       "35                  http://distilldeep.ucd.ie/paleale/   \n",
       "36                                               Keras   \n",
       "37                                               Keras   \n",
       "38                                   Keras, TensorFlow   \n",
       "39          Pytorch, https://embed.protein.properties/   \n",
       "40                                                 NaN   \n",
       "41                                                 NaN   \n",
       "42   TensorFlow, https://github.com/yfCuiFaith/Deep...   \n",
       "43   TensorFlow,\\nhttps://sparks-lab.org/server/spr...   \n",
       "44   Keras, Tensorflow, https://\\ngithub.com/GYBTA/...   \n",
       "45                                                 NaN   \n",
       "46                              Zhang and Quark server   \n",
       "47           AIR 2.0 www.csbio.sjtu.edu.cn/bioinf/AIR/   \n",
       "48   TensorFlow,    DeepUbi: https://github.com/Sun...   \n",
       "49   TensorFlow ,  Ornate  https://team.inria.fr/na...   \n",
       "50             N/A    https://simtk.org/projects/fscnn   \n",
       "51                     Magenta TensorFlow,  Melody RNN   \n",
       "52   TensorFlow, https://www.eecis.udel.edu/~lliao/...   \n",
       "53                                                 NaN   \n",
       "54      http://sparks-lab.org/jack/server/SPOTContact/   \n",
       "55   Keras, Theano,\\nhttps://github.com/jiagenlee/D...   \n",
       "56                      http://sparks-lab.org.\\n\\nSPIN   \n",
       "57          Keras,\\nhttps://github.com/lje00006/DeepT3   \n",
       "58                              PyTorch, NumPy & SciPy   \n",
       "59                                          TensorFlow   \n",
       "60                                               Keras   \n",
       "61            https://\\ngithub.com/USTCHIlab/DeepPhos.   \n",
       "62                                       MXNet ,   N/A   \n",
       "63                                  COSNet,  ParCOSNet   \n",
       "64                                     TensorFlow, N/A   \n",
       "65   Keras, Theano     DeepRab;  http://bio216.bioi...   \n",
       "66           Gnina ,  Caffe,  https://github.com/gnina   \n",
       "67   Keras, Theano,  https  ://github.com/Seeliglab...   \n",
       "68                    Keras, Theano,  TensorFlow , N/A   \n",
       "69   PyTorch, TensorFlow    ReLeaSE https://github....   \n",
       "70                                                 NaN   \n",
       "71           DisEMBL, IUPred, VSL2, DBann, and Espritz   \n",
       "72                   Gnina , Caffe    Github.com/gnina   \n",
       "73   PROSO II  https://zenodo.org/record/1162886#.X...   \n",
       "74   SciPy     https://github.  com/stamakro/SSP-LSDR.   \n",
       "75                     MUFOLD-SS\\nTensorFlow and Keras   \n",
       "76   TensorFlow, Keras \\nhttp://sysbio.rnet.missour...   \n",
       "77   Lasagne, Theano,   Deep Loc: http://www.cbs.dt...   \n",
       "78               Keras, Theano    www.playmolecule.org   \n",
       "79   TensorFlow,  Theano    http://raptorx.uchicago...   \n",
       "80   Keras, Theano,   https://github.com/uci-cbcl/H...   \n",
       "81   modlAMP  Python package    https://github.com/...   \n",
       "82                                          TensorFlow   \n",
       "83   SECLAF,  TensorFlow    https://pitgroup.org/se...   \n",
       "84                               ProLanGO Model    N/A   \n",
       "85            Gnina  Caffe    https://github.com/gnina   \n",
       "86          ELFIN    https://github.com/joy13975/elfin   \n",
       "87                                                 NaN   \n",
       "88   CYSCON    http://www.csbio.sjtu.edu.cn/bioinf/...   \n",
       "89   MDLoc    http://www.eecis.udel.edu/compbio/mdloc.   \n",
       "90                               MATLAB  Bioweka  Weka   \n",
       "91                                                 NaN   \n",
       "92                                                 NaN   \n",
       "93   EASE-MM  LISVM    http://www.ict.griffigr.edu....   \n",
       "94                      SPIN     http://sparks-lab.org   \n",
       "95                                                 NaN   \n",
       "96   kmerHMM    http://www.cs.toronto.edu/wkc/  kme...   \n",
       "97   CPD    http://genoweb.toulouse.inra.fr/tschiex...   \n",
       "98                                              Torch5   \n",
       "99            http://pubs.acs.org    Modeller, Rosetta   \n",
       "100  SMURFLite    http://smurf.cs.tufts.edu/smurflite/   \n",
       "101  NNScore  2.0    http://www.nbcr.net/software/n...   \n",
       "102   DROP    http://web.tuat.ac.jp/~domserv/DROP.html   \n",
       "103                      http://tau.ac.il/~haimash/WMC   \n",
       "104  SPARKS-X    https://sparks-lab.org/server/spar...   \n",
       "105  CABS- NMR  toolkit    http://biocomp.chem.uw.e...   \n",
       "106                                                CNF   \n",
       "107                YLoc  Weka    www.multiloc.org/YLoc   \n",
       "108  SeqRate    http://casp.rnet.missouri.edu/fold_...   \n",
       "109  Prethermut     http://www.mobioinfor.cn/prethe...   \n",
       "110                                                NaN   \n",
       "111                                             BroMAP   \n",
       "112  MOSEK    http://www.prc.boun.edu.tr/appserv/pr...   \n",
       "113                              CPC-X Software    N/A   \n",
       "114                                                NaN   \n",
       "115  LIBSVM  ProlocGO    http://iclab.life.nctu.edu...   \n",
       "116  MUSTER   http://zhang.bioinformatics.ku.edu/MU...   \n",
       "117  PREDICT-2ND    http://www.soe.ucsc.edu/∼karplu...   \n",
       "118  SVM-HUSTLE    http://www.sysbio.org/sysbio/net...   \n",
       "119                                                NaN   \n",
       "120                                     LIBSVM-library   \n",
       "121                                                NaN   \n",
       "122                                                NaN   \n",
       "123  LIBSVM-library     DHPRED  http://www.fz-  jue...   \n",
       "124                                               BBFN   \n",
       "125                                                NaN   \n",
       "126  I-Mutant2.0     http://gpcr.biocomp.unibo.it/c...   \n",
       "127                                                NaN   \n",
       "128                                                NaN   \n",
       "129                                                NaN   \n",
       "130                                                NaN   \n",
       "131                                                NaN   \n",
       "132                                                NaN   \n",
       "\n",
       "                                           7  \\\n",
       "0            Programming Language / Platform   \n",
       "1                                        NaN   \n",
       "2                                     Python   \n",
       "3                                     Python   \n",
       "4                                     Python   \n",
       "5                                     Python   \n",
       "6                       Matlab, Java, Python   \n",
       "7                                     Python   \n",
       "8                                        NaN   \n",
       "9                                     Python   \n",
       "10                                    Python   \n",
       "11                                       NaN   \n",
       "12                        Multiple languages   \n",
       "13                                    Python   \n",
       "14                                       NaN   \n",
       "15                                    Python   \n",
       "16                                    Python   \n",
       "17                                         C   \n",
       "18                                       NaN   \n",
       "19                             C++, \\nPython   \n",
       "20                                    Python   \n",
       "21                                       NaN   \n",
       "22                                    Python   \n",
       "23                                    Python   \n",
       "24                                    Matlab   \n",
       "25                                    Python   \n",
       "26                                    Python   \n",
       "27                                    Python   \n",
       "28                                    Python   \n",
       "29                                    Python   \n",
       "30                                    Python   \n",
       "31                                       NaN   \n",
       "32                                       NaN   \n",
       "33                                    Python   \n",
       "34                                    Python   \n",
       "35                                    Python   \n",
       "36                                    Python   \n",
       "37                                    Python   \n",
       "38                                    Python   \n",
       "39                                    Python   \n",
       "40                                       NaN   \n",
       "41                                       NaN   \n",
       "42                                    Python   \n",
       "43                                    Python   \n",
       "44                                    Python   \n",
       "45                                       NaN   \n",
       "46                                       NaN   \n",
       "47                                    Python   \n",
       "48                    Python, Matlab,\\nLinux   \n",
       "49                            C++,  \\nPython   \n",
       "50                                    Python   \n",
       "51                            Java, \\nPython   \n",
       "52                                    Python   \n",
       "53                                       NaN   \n",
       "54                                       NaN   \n",
       "55                                    Python   \n",
       "56                                       NaN   \n",
       "57                                    Python   \n",
       "58                                    Python   \n",
       "59                                    Python   \n",
       "60                                    Python   \n",
       "61                                       NaN   \n",
       "62                                         R   \n",
       "63                           C, C++, R, CUDA   \n",
       "64                          C, C++, \\nPython   \n",
       "65                                    Python   \n",
       "66                             C++, \\nPython   \n",
       "67                                    Python   \n",
       "68                                    Python   \n",
       "69                            Python, \\nCUDA   \n",
       "70                                       NaN   \n",
       "71                                       NaN   \n",
       "72                               C++, Python   \n",
       "73                         Python,\\n \\nLinux   \n",
       "74   Python, Matlab   Bioinformatics toolbox   \n",
       "75                                    Python   \n",
       "76                                    Python   \n",
       "77                                    Python   \n",
       "78                                    Python   \n",
       "79                                    Python   \n",
       "80                                    Python   \n",
       "81                                    Python   \n",
       "82                                    Python   \n",
       "83                                    Python   \n",
       "84                                       NaN   \n",
       "85                                       C++   \n",
       "86                       Python, C++, Matlab   \n",
       "87                                       NaN   \n",
       "88                                       NaN   \n",
       "89                                    Python   \n",
       "90                              Java, Matlab   \n",
       "91                                       NaN   \n",
       "92                                       NaN   \n",
       "93                             Python, Linux   \n",
       "94                            Python , Linux   \n",
       "95                                       NaN   \n",
       "96                                       NaN   \n",
       "97                                      Perl   \n",
       "98                                         C   \n",
       "99                                    Python   \n",
       "100                                      NaN   \n",
       "101                                   Python   \n",
       "102                              Bash script   \n",
       "103                                     Perl   \n",
       "104                             Shell script   \n",
       "105                                      NaN   \n",
       "106                                      NaN   \n",
       "107                      Python, Java, Linux   \n",
       "108                                     Java   \n",
       "109                          R, Perl,  Linux   \n",
       "110                                      NaN   \n",
       "111                               C++, Linux   \n",
       "112                                      NaN   \n",
       "113                                     Java   \n",
       "114                                      NaN   \n",
       "115                                      NaN   \n",
       "116                                      NaN   \n",
       "117                                      C++   \n",
       "118                                      NaN   \n",
       "119                                      C++   \n",
       "120     C++, Java , Python    Windows ,Linux   \n",
       "121                                   Matlab   \n",
       "122                                   Matlab   \n",
       "123               C, Python, Linux,  Windows   \n",
       "124                                      NaN   \n",
       "125                                      NaN   \n",
       "126                                   Python   \n",
       "127                                      NaN   \n",
       "128                                      NaN   \n",
       "129                                      NaN   \n",
       "130                                      NaN   \n",
       "131                                      NaN   \n",
       "132                                        C   \n",
       "\n",
       "                                          8            9  \\\n",
       "0                                   License  Quality (%)   \n",
       "1                                       NaN           70   \n",
       "2                               Open Source           80   \n",
       "3                               Open Source           75   \n",
       "4                               Open Source          100   \n",
       "5                               Open Source           85   \n",
       "6                               Open Source           90   \n",
       "7                               Open Source           95   \n",
       "8                              Open\\nSource           60   \n",
       "9                               Open Source           90   \n",
       "10                             Open\\nSource           70   \n",
       "11                                      NaN           60   \n",
       "12                                      NaN           60   \n",
       "13                            Open \\nSource           70   \n",
       "14                             Open\\nSource           65   \n",
       "15                              Open Source          100   \n",
       "16                              Open Source           90   \n",
       "17                              Open source           60   \n",
       "18                                      NaN           50   \n",
       "19                              Open Source           60   \n",
       "20                                      NaN          100   \n",
       "21                                      NaN           75   \n",
       "22                              Open Source           70   \n",
       "23                              Open source          100   \n",
       "24                                MathWorks           90   \n",
       "25                              Open Source           90   \n",
       "26                              Open Source           80   \n",
       "27                              Open Source           70   \n",
       "28                              Open Source           70   \n",
       "29                              Open Source           75   \n",
       "30                             Open\\nSource           70   \n",
       "31                                      NaN           60   \n",
       "32                                      NaN           60   \n",
       "33                             Open\\nSource           60   \n",
       "34                              Open Source           85   \n",
       "35                              Open Source           90   \n",
       "36                              Open source          100   \n",
       "37                              Open Source           50   \n",
       "38                              Open Source          100   \n",
       "39                              Open Source           80   \n",
       "40                                      NaN           75   \n",
       "41                                      NaN           75   \n",
       "42                              Open Source          100   \n",
       "43                              Open Source           80   \n",
       "44                              Open Source           80   \n",
       "45                                      NaN           60   \n",
       "46                              Open Source           85   \n",
       "47                              Open Source           95   \n",
       "48                              Open Source          100   \n",
       "49                              Open Source           85   \n",
       "50                                      NaN           75   \n",
       "51                              Open Source          100   \n",
       "52                              Open Source           75   \n",
       "53                                      NaN           80   \n",
       "54                                      NaN           95   \n",
       "55                              Open Source           80   \n",
       "56                              Open Source           65   \n",
       "57                              Open Source           60   \n",
       "58                              Open Source          100   \n",
       "59                              Open Source          100   \n",
       "60                              Open Source          100   \n",
       "61                                      NaN           60   \n",
       "62                              Apache  2.0           85   \n",
       "63                              Open Source           75   \n",
       "64                              Apache  2.0           65   \n",
       "65                              Open Source           90   \n",
       "66                              Open Source          100   \n",
       "67                              Open Source           85   \n",
       "68                              Open Source           85   \n",
       "69                              Open Source           75   \n",
       "70                                      NaN           50   \n",
       "71                                      NaN           50   \n",
       "72                              Open source           50   \n",
       "73                             Open  source           95   \n",
       "74                  Open  source, Mathworks           80   \n",
       "75                              Open Source           80   \n",
       "76                              Open Source           65   \n",
       "77                              License MIT           90   \n",
       "78                              Open Source           90   \n",
       "79                              Apache  2.0           75   \n",
       "80                              Open Source          100   \n",
       "81                              Open Source          100   \n",
       "82                              Open Source           90   \n",
       "83                              Open Source           85   \n",
       "84                                      NaN           50   \n",
       "85                              Open Source           85   \n",
       "86   Apache 2.0 open source    3-Clause BSD           90   \n",
       "87                                      NaN           50   \n",
       "88                                      NaN           75   \n",
       "89                              Open Source           75   \n",
       "90                  Open Source   MathWorks           50   \n",
       "91                                      NaN           50   \n",
       "92                                      NaN           50   \n",
       "93                              Open Source           75   \n",
       "94                              Open Source           85   \n",
       "95                                      NaN           65   \n",
       "96                                      NaN           50   \n",
       "97                             Open  source           65   \n",
       "98                              Open Source          100   \n",
       "99         IBM, Academic nonprofit freeware           75   \n",
       "100                             Open Source           65   \n",
       "101                             Open Source           50   \n",
       "102                            Open  source           75   \n",
       "103                             Open Source           45   \n",
       "104                             Open Source           75   \n",
       "105                                     NaN           70   \n",
       "106                                     NaN           80   \n",
       "107                            Open  source           85   \n",
       "108                             Open Source           75   \n",
       "109                             Open Source           75   \n",
       "110                                     NaN           50   \n",
       "111                           Open   Source          100   \n",
       "112                            Open  Source           85   \n",
       "113                Neural Power version 2.5           75   \n",
       "114                                     NaN           65   \n",
       "115                                     NaN           75   \n",
       "116                                     NaN           50   \n",
       "117                             Open source           80   \n",
       "118                                     NaN           70   \n",
       "119                            Open  source           50   \n",
       "120                           Open   source           80   \n",
       "121                               MathWorks           75   \n",
       "122                               MathWorks           75   \n",
       "123                            Open  source           80   \n",
       "124                                     NaN           75   \n",
       "125                                     NaN           75   \n",
       "126                             Open Source           75   \n",
       "127                                    N/A.           65   \n",
       "128                                     NaN           50   \n",
       "129                                     NaN           45   \n",
       "130                                     NaN           65   \n",
       "131                                     NaN           90   \n",
       "132                                     NaN           50   \n",
       "\n",
       "                                                    10  \\\n",
       "0                                     Machine Learning   \n",
       "1                 Supervised Learning:\\nClassification   \n",
       "2    Supervised Learning:\\nRegression\\nPrediction\\n...   \n",
       "3                     Supervised Learning:\\nPrediction   \n",
       "4                           Supervised learning models   \n",
       "5                     Supervised Learning:\\nPrediction   \n",
       "6                     Supervised Learning:\\nPrediction   \n",
       "7         Supervised Learning:\\nPrediction\\nRegression   \n",
       "8                                                  NaN   \n",
       "9    Supervised Learning;\\nUnsupervised Learning;\\n...   \n",
       "10                   Supervised Learning: \\nRegression   \n",
       "11   Supervised Learning, Semi-Supervised Learning:...   \n",
       "12         Supervised Learning;\\nUnsupervised Learning   \n",
       "13             Supervised Learning:\\nClass\\nPrediction   \n",
       "14     Supervised Learning: Regression\\nClassification   \n",
       "15                     Supervised Learning, Prediction   \n",
       "16                     Supervised Learning Prediction.   \n",
       "17                      Supervised Learning Prediction   \n",
       "18                     Supervised Learning, Classifier   \n",
       "19                     Supervised Learning, Prediction   \n",
       "20                 Supervised Learning: Classification   \n",
       "21                     Supervised learning: Prediction   \n",
       "22                 Supervised Learning: Classification   \n",
       "23                 Supervised Learning: Classification   \n",
       "24                    Supervised Learning:\\nPrediction   \n",
       "25                Supervised Learning:\\nClassification   \n",
       "26   Supervised Learning: \\nPrediction\\nClassification   \n",
       "27        Supervised Learning:\\nRegression\\nPrediction   \n",
       "28                    Supervised Learning:\\nPrediction   \n",
       "29                   Supervised Learning: \\nRegression   \n",
       "30                    Supervised Learning:\\nRegression   \n",
       "31        Supervised Learning:\\nRegression\\nPrediction   \n",
       "32   Supervised Learning, Unsupervised Learning: \\n...   \n",
       "33   Supervised Learning;\\nUnsupervised Learning;\\n...   \n",
       "34                 Supervised Learning: Classification   \n",
       "35                     Supervised Learning, Prediction   \n",
       "36                     Supervised Learning, Prediction   \n",
       "37                          Classification of articles   \n",
       "38                     Supervised Learning: Prediction   \n",
       "39                 Supervised Learning: Classification   \n",
       "40                 Supervised Learning: Classification   \n",
       "41                 Supervised Learning: Classification   \n",
       "42                     Supervised Learning: Prediction   \n",
       "43       Supervised Learning: \\nRegression\\nPrediction   \n",
       "44   Supervised Learning: \\nPrediction\\nClassification   \n",
       "45                 Supervised Learning: Classification   \n",
       "46     Supervised Learning: Classification\\nRegression   \n",
       "47                   Supervised Learning: Optimization   \n",
       "48                 Supervised Learning: Classification   \n",
       "49                     Supervised Learning: Regression   \n",
       "50                 Supervised Learning: Classification   \n",
       "51                     Supervised Learning: Regression   \n",
       "52                    Supervised Learning:\\nPrediction   \n",
       "53                 Supervised Learning: Classification   \n",
       "54                    Supervised Learning:\\nPrediction   \n",
       "55   Supervised Learning:\\nRegression\\nClassificati...   \n",
       "56                   Supervised Learning: \\nPrediction   \n",
       "57   Supervised Learning: \\nRegression\\nClassification   \n",
       "58                     Supervised Learning, Prediction   \n",
       "59                     Supervised Learning, Prediction   \n",
       "60                     Supervised Learning, Prediction   \n",
       "61      Supervised Learning: \\nRegression\\nPredicition   \n",
       "62                 Supervised Learning: Classification   \n",
       "63                   Unsupervised Learning: Clustering   \n",
       "64                     Supervised Learning: Regression   \n",
       "65                     Supervised Learning: Regression   \n",
       "66                     Supervised Learning: Regression   \n",
       "67                     Supervised Learning: Regression   \n",
       "68                     Supervised Learning: Regression   \n",
       "69   Reinforced Learning, Unsupervised Learning: Re...   \n",
       "70                 Supervised Learning: Classification   \n",
       "71                     Supervised Learning: Regression   \n",
       "72                 Supervised Learning: Classification   \n",
       "73                   Unsupervised Learning: Clustering   \n",
       "74                  Unsupervised Learning:  Clustering   \n",
       "75                 Supervised Learning: Classification   \n",
       "76       Supervised Learning:\\nRegression\\nPredecition   \n",
       "77                 Supervised Learning: Classification   \n",
       "78                     Supervised Learning: Regression   \n",
       "79      Supervised Learning: Classification Regression   \n",
       "80                     Supervised Learning: Regression   \n",
       "81                     Supervised Learning: Regression   \n",
       "82                 Supervised Learning: Classification   \n",
       "83                 Supervised Learning: Classification   \n",
       "84                 Supervised Learning: Classification   \n",
       "85                 Supervised Learning: Classification   \n",
       "86                   Supervised Learning: Optimization   \n",
       "87                 Supervised Learning: Classification   \n",
       "88                     Supervised Learning: Regression   \n",
       "89                 Supervised Learning: Classification   \n",
       "90                 Supervised Learning: Classification   \n",
       "91                                                 NaN   \n",
       "92                                                 NaN   \n",
       "93                 Supervised Learning: Classification   \n",
       "94                 Supervised Learning: Classification   \n",
       "95                 Supervised Learning: Classification   \n",
       "96   Supervised Learning: Classification.  Unsuperv...   \n",
       "97                 Supervised Learning: Classification   \n",
       "98                 Supervised Learning: Classification   \n",
       "99                                                 NaN   \n",
       "100                 Unsupervised Learning:  Clustering   \n",
       "101                Supervised Learning: Classification   \n",
       "102                Supervised Learning: Classification   \n",
       "103                                                NaN   \n",
       "104                    Supervised Learning: Regression   \n",
       "105                  Unsupervised Learning: Clustering   \n",
       "106                Supervised Learning: Classification   \n",
       "107                Supervised Learning: Classification   \n",
       "108                Supervised Learning: Classification   \n",
       "109                Supervised Learning: Classification   \n",
       "110                Supervised Learning: Classification   \n",
       "111                Supervised Learning: Classification   \n",
       "112     Supervised Learning: Classification Regression   \n",
       "113                Supervised Learning: Classification   \n",
       "114                                                NaN   \n",
       "115                Supervised Learning: Classification   \n",
       "116                Supervised Learning: Classification   \n",
       "117                  Unsupervised Learning: Clustering   \n",
       "118                Supervised Learning: Classification   \n",
       "119                                                NaN   \n",
       "120                Supervised Learning: Classification   \n",
       "121     Supervised Learning: Classification Regression   \n",
       "122                  Unsupervised Learning: Clustering   \n",
       "123                Supervised Learning: Classification   \n",
       "124                    Supervised Learning: Regression   \n",
       "125                Supervised Learning: Classification   \n",
       "126                Supervised Learning: Classification   \n",
       "127                Supervised Learning: Classification   \n",
       "128                 Unsupervised Learning:  Clustering   \n",
       "129                    Supervised Learning: Regression   \n",
       "130                                                NaN   \n",
       "131                    Supervised Learning: Regression   \n",
       "132                Supervised Learning: Classification   \n",
       "\n",
       "                                                    11  \n",
       "0                                  Protein Application  \n",
       "1                               Protein classification  \n",
       "2           Prediction of Protein–ATP Binding Residues  \n",
       "3                Protein ligand interaction prediction  \n",
       "4                           Physicochemical properties  \n",
       "5                  Protein affinity-binding prediction  \n",
       "6                    Protein-protein binding affinity   \n",
       "7                                       Protein design  \n",
       "8                        Protein domain identification  \n",
       "9                                       Protein Design  \n",
       "10                             Contact map\\nprediction  \n",
       "11                        Drug-target binding affinity  \n",
       "12                           Drug development o design  \n",
       "13            Microenvironment mutation identification  \n",
       "14                   Protein-ligand binding prediction  \n",
       "15                        Protein structure prediction  \n",
       "16            Protein function prediction, Function ID  \n",
       "17                                      Protein design  \n",
       "18                              Human genes prediction  \n",
       "19                        Protein structure prediction  \n",
       "20                                         Function ID  \n",
       "21                                         Function ID  \n",
       "22                         Protein function prediction  \n",
       "23                                             Fold ID  \n",
       "24                                         Drug design  \n",
       "25                                      Protein design  \n",
       "26                         Protein function prediction  \n",
       "27                          Protein binding prediction  \n",
       "28                        Protein structure prediction  \n",
       "29          protein\\nsub-mitochondrial site prediction  \n",
       "30                         Protein function prediction  \n",
       "31                       Protein solubility prediction  \n",
       "32                 Interface residue\\npairs prediction  \n",
       "33                        Protein structure prediction  \n",
       "34                             Function ID and Fold ID  \n",
       "35   Protein relative solvent accessibility  predic...  \n",
       "36                        Protein structure prediction  \n",
       "37                          Physicochemical properties  \n",
       "38                             Protein site prediction  \n",
       "39                                             Fold ID  \n",
       "40                          Physicochemical properties  \n",
       "41                             Function ID and Fold ID  \n",
       "42                             Protein site prediction  \n",
       "43                  Glycosylation\\nsite identification  \n",
       "44              Protein secondary structure prediction  \n",
       "45                            Glycation site predictor  \n",
       "46                        Protein structure prediction  \n",
       "47                        Protein structure prediction  \n",
       "48                             Protein site prediction  \n",
       "49                            Model protein prediction  \n",
       "50                         Protein function prediction  \n",
       "51                         Protein sequence prediction  \n",
       "52                         Protein-protein interaction  \n",
       "53                                         Function ID  \n",
       "54                      Protein contact-map prediction  \n",
       "55                         Acetylation site prediction  \n",
       "56                      Protein  sequence prediction\\n  \n",
       "57                              Protein classification  \n",
       "58                         Protein function prediction  \n",
       "59                         Protein function prediction  \n",
       "60                        Protein structure prediction  \n",
       "61                      Phosphorylation site predictor  \n",
       "62                         Protein function prediction  \n",
       "63                         Protein function prediction  \n",
       "64                                         Drug Design  \n",
       "65                              Protein Classification  \n",
       "66                            Protein model prediction  \n",
       "67                                             Fold ID  \n",
       "68                         Protein function prediction  \n",
       "69                                         Drug Design  \n",
       "70                                       Enzyme Design  \n",
       "71         Intrinsically disordered protein prediction  \n",
       "72                                     Protein Scoring  \n",
       "73                         Protein function prediction  \n",
       "74                         Protein function prediction  \n",
       "75                        Protein structure prediction  \n",
       "76                    Protein residue–residue contacts  \n",
       "77                        Protein structure prediction  \n",
       "78                                         Drug Design  \n",
       "79                        Protein structure prediction  \n",
       "80                        Protein structure prediction  \n",
       "81                                         Drug Design  \n",
       "82                              Protein Classification  \n",
       "83                              Protein Classification  \n",
       "84                         Protein function prediction  \n",
       "85                                         Drug Design  \n",
       "86                                         Drug Design  \n",
       "87                         Protein function prediction  \n",
       "88                        Protein structure prediction  \n",
       "89                             Protein site prediction  \n",
       "90                                         Drug Design  \n",
       "91                                         Drug Design  \n",
       "92                                         Drug Design  \n",
       "93                                        Model Design  \n",
       "94                        Protein structure prediction  \n",
       "95                                         Drug Design  \n",
       "96                                     Model Discovery  \n",
       "97                                      Protein design  \n",
       "98                         Protein function prediction  \n",
       "99                                         Drug Design  \n",
       "100                                        Drug Design  \n",
       "101                             Protein map prediction  \n",
       "102                                        Drug Design  \n",
       "103                             Protein map prediction  \n",
       "104                        Protein function prediction  \n",
       "105                        Protein function prediction  \n",
       "106                                            Fold ID  \n",
       "107                            Protein site prediction  \n",
       "108                             Protein map prediction  \n",
       "109                        Protein function prediction  \n",
       "110                                       Model Design  \n",
       "111                                        Drug Design  \n",
       "112                                       Model Design  \n",
       "113                                     Protein design  \n",
       "114                                        Drug Design  \n",
       "115                            Protein site prediction  \n",
       "116                        Protein function prediction  \n",
       "117                       Protein structure prediction  \n",
       "118                            Function ID and Fold ID  \n",
       "119                                            Fold ID  \n",
       "120                                            Fold ID  \n",
       "121                                     Protein design  \n",
       "122                                     Protein design  \n",
       "123                                     Protein design  \n",
       "124                        Protein function prediction  \n",
       "125                        Protein function prediction  \n",
       "126                        Protein function prediction  \n",
       "127                        Protein function prediction  \n",
       "128                        Protein function prediction  \n",
       "129                       Protein structure prediction  \n",
       "130                                     Protein design  \n",
       "131                                        Drug Design  \n",
       "132                            Protein site prediction  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "altered-terrorist",
   "metadata": {
    "id": "altered-terrorist"
   },
   "outputs": [],
   "source": [
    "df=df[:78]  # otra forma chida con un for y list comprehension df.drop(labels=[i for i in range(78,103)], axis=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "stainless-directory",
   "metadata": {
    "id": "stainless-directory"
   },
   "source": [
    "---\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "spanish-synthesis",
   "metadata": {
    "id": "spanish-synthesis"
   },
   "source": [
    "# Lenguaje Python\n",
    "* En la celda de abajo buscamos en la columna de Programming Language todos los que contenga la palabra Python\n",
    "de esta forma para poder identificar cuantos artículos usar python. Además, se muestran los autores de correspondencia.\n",
    "* Los resultado se compararon con la tabla de excely muestra los mismos resultados para el lenguaje python."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "weird-seafood",
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 1000
    },
    "id": "weird-seafood",
    "outputId": "810dfb05-d5b9-4cd3-89c8-b41aa891b740"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True     70\n",
      "False    63\n",
      "Name: 7, dtype: int64\n",
      "\n",
      "True     70\n",
      "NaN      41\n",
      "False    22\n",
      "Name: 7, dtype: int64\n",
      "\n",
      "(70, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>1</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>Song J., 2021, \\nChina</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>Verma N., 2021, \\nUS</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Li H., 2020, \\nFrance, \\nHong Kong,</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Kwon Y., \\n2020, Korea</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>Wang M.., 2020, US</td>\n",
       "      <td>Matlab, Java, Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>7</td>\n",
       "      <td>Qin Z., \\n2020, US</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>9</td>\n",
       "      <td>Yu C., 2020, \\nTaiwan, US</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>10</td>\n",
       "      <td>Yang H., 2020, \\nChina</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>13</td>\n",
       "      <td>Shroff R., 2020, \\nUS</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>15</td>\n",
       "      <td>Wardah W., 2020, \\nAustralia, Fiji, Japan, US</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>16</td>\n",
       "      <td>Liang M., \\n2020, \\nChina</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>19</td>\n",
       "      <td>Bond P., 2020, \\nUK</td>\n",
       "      <td>C++, \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>20</td>\n",
       "      <td>Nie J., 2020, Singapore, Taiwan</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>22</td>\n",
       "      <td>Du Z., 2020, \\nChina, Russia, US</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>23</td>\n",
       "      <td>Li C., 2019, \\nChina</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>25</td>\n",
       "      <td>Fang C., 2019, US</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>26</td>\n",
       "      <td>Wan C., 2019, \\nUK</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>27</td>\n",
       "      <td>Luo X., 2019, \\nChina</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>28</td>\n",
       "      <td>Mirabello C., 2019, \\nSweden</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>29</td>\n",
       "      <td>Savojardo C., 2019, \\nItaly</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>30</td>\n",
       "      <td>Rifaioglu A., 2019, \\nTurkey, UK</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>33</td>\n",
       "      <td>Long S., 2019, \\nChina</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>34</td>\n",
       "      <td>Zhang Y., China, 2019</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>35</td>\n",
       "      <td>Kaleel M., 2019, \\nIreland</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>36</td>\n",
       "      <td>Schantz M., 2019, \\nArgentina, \\nDenmark, \\nMa...</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>37</td>\n",
       "      <td>Luo L., \\n2019, \\nChina</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>38</td>\n",
       "      <td>Haberal I., 2019, Norway, Turkey</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>39</td>\n",
       "      <td>Heinzinger M., 2019, \\nGermany</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>42</td>\n",
       "      <td>Cui Y., \\n2019, \\nChina\\n</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>43</td>\n",
       "      <td>Taherzadeh G., 2019, Australia, US</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>44</td>\n",
       "      <td>Guo Y., 2019, US</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>47</td>\n",
       "      <td>Wang D., 2019, China</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>48</td>\n",
       "      <td>Fu H.,\\n2019, \\nChina</td>\n",
       "      <td>Python, Matlab,\\nLinux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>49</td>\n",
       "      <td>Pagès G., \\n2019, \\nFrance</td>\n",
       "      <td>C++,  \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>50</td>\n",
       "      <td>Torng  W., \\n2019,  \\nUS</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>51</td>\n",
       "      <td>Yu C., \\n2019, \\nUS</td>\n",
       "      <td>Java, \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>52</td>\n",
       "      <td>Huang L., 2018, US</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>55</td>\n",
       "      <td>Zhao X., 2018, \\nChina</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>57</td>\n",
       "      <td>Xue L., \\n2018, \\nChina, US</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>58</td>\n",
       "      <td>Feinberg E., \\n2018, \\nChina, US</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>59</td>\n",
       "      <td>Zhang L.,\\n2018, \\nChina</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>60</td>\n",
       "      <td>Zhang B., 2018, \\nChina</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>64</td>\n",
       "      <td>Hu B., \\n2018, \\nChina</td>\n",
       "      <td>C, C++, \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>65</td>\n",
       "      <td>Le N., \\n2018, \\nTaiwan</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>66</td>\n",
       "      <td>Sunseri J., 2018, \\nUS</td>\n",
       "      <td>C++, \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>67</td>\n",
       "      <td>Cuperus J., 2018, \\nUS</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>68</td>\n",
       "      <td>Li H., \\n2018, \\nChina</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>69</td>\n",
       "      <td>Popova M., 2018, \\nRussia, US</td>\n",
       "      <td>Python, \\nCUDA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>72</td>\n",
       "      <td>Hochuli J., \\n2018, \\nUS</td>\n",
       "      <td>C++, Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>73</td>\n",
       "      <td>Khurana S., 2018, Qatar, US</td>\n",
       "      <td>Python,\\n \\nLinux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>74</td>\n",
       "      <td>Makrodimitris S., \\n2018,  Netherlands</td>\n",
       "      <td>Python, Matlab   Bioinformatics toolbox</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>75</td>\n",
       "      <td>Fang C., 2018, \\nUS</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>76</td>\n",
       "      <td>Adhikari B., 2017,\\nUS</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>77</td>\n",
       "      <td>Armenteros J., 2017, \\nDenmark</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>78</td>\n",
       "      <td>Jimenez J., \\n2017, \\nSpain</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>79</td>\n",
       "      <td>Wang S., \\n2017, \\nUS</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>80</td>\n",
       "      <td>Vang Y., \\n2017, \\nUS</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>81</td>\n",
       "      <td>Müller A., 2017, Switzerland</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>82</td>\n",
       "      <td>Szalkai B., 2017,\\nHungary</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>83</td>\n",
       "      <td>Szalkai B., \\n2017, \\nHungary</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>86</td>\n",
       "      <td>Yeh C., \\n2017, \\nUK, US</td>\n",
       "      <td>Python, C++, Matlab</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>89</td>\n",
       "      <td>Simha R., 2015, \\nCanada, Germany, US</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>93</td>\n",
       "      <td>Folkman L., \\n2014, \\nAustralia</td>\n",
       "      <td>Python, Linux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>94</td>\n",
       "      <td>Li Z., \\n2014, \\nUS</td>\n",
       "      <td>Python , Linux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>99</td>\n",
       "      <td>Eisenbeis S., \\n2012, \\nGermany</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>101</td>\n",
       "      <td>Durrant J.,  2011, \\nUS</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>107</td>\n",
       "      <td>Briesemeister S., \\n2010, \\nGermany</td>\n",
       "      <td>Python, Java, Linux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>120</td>\n",
       "      <td>Shamim M., \\n2007, \\nIndia</td>\n",
       "      <td>C++, Java , Python    Windows ,Linux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>123</td>\n",
       "      <td>Zimmermann O., \\n2006, \\nGermany</td>\n",
       "      <td>C, Python, Linux,  Windows</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>126</td>\n",
       "      <td>Capriotti E., 2005, \\nItaly</td>\n",
       "      <td>Python</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    index                                                  1  \\\n",
       "0       2                             Song J., 2021, \\nChina   \n",
       "1       3                               Verma N., 2021, \\nUS   \n",
       "2       4               Li H., 2020, \\nFrance, \\nHong Kong,    \n",
       "3       5                             Kwon Y., \\n2020, Korea   \n",
       "4       6                                 Wang M.., 2020, US   \n",
       "5       7                                 Qin Z., \\n2020, US   \n",
       "6       9                          Yu C., 2020, \\nTaiwan, US   \n",
       "7      10                             Yang H., 2020, \\nChina   \n",
       "8      13                              Shroff R., 2020, \\nUS   \n",
       "9      15      Wardah W., 2020, \\nAustralia, Fiji, Japan, US   \n",
       "10     16                          Liang M., \\n2020, \\nChina   \n",
       "11     19                                Bond P., 2020, \\nUK   \n",
       "12     20                    Nie J., 2020, Singapore, Taiwan   \n",
       "13     22                   Du Z., 2020, \\nChina, Russia, US   \n",
       "14     23                               Li C., 2019, \\nChina   \n",
       "15     25                                  Fang C., 2019, US   \n",
       "16     26                                 Wan C., 2019, \\nUK   \n",
       "17     27                              Luo X., 2019, \\nChina   \n",
       "18     28                       Mirabello C., 2019, \\nSweden   \n",
       "19     29                        Savojardo C., 2019, \\nItaly   \n",
       "20     30                   Rifaioglu A., 2019, \\nTurkey, UK   \n",
       "21     33                             Long S., 2019, \\nChina   \n",
       "22     34                              Zhang Y., China, 2019   \n",
       "23     35                         Kaleel M., 2019, \\nIreland   \n",
       "24     36  Schantz M., 2019, \\nArgentina, \\nDenmark, \\nMa...   \n",
       "25     37                            Luo L., \\n2019, \\nChina   \n",
       "26     38                   Haberal I., 2019, Norway, Turkey   \n",
       "27     39                     Heinzinger M., 2019, \\nGermany   \n",
       "28     42                          Cui Y., \\n2019, \\nChina\\n   \n",
       "29     43                 Taherzadeh G., 2019, Australia, US   \n",
       "30     44                                   Guo Y., 2019, US   \n",
       "31     47                               Wang D., 2019, China   \n",
       "32     48                              Fu H.,\\n2019, \\nChina   \n",
       "33     49                         Pagès G., \\n2019, \\nFrance   \n",
       "34     50                           Torng  W., \\n2019,  \\nUS   \n",
       "35     51                                Yu C., \\n2019, \\nUS   \n",
       "36     52                                 Huang L., 2018, US   \n",
       "37     55                             Zhao X., 2018, \\nChina   \n",
       "38     57                        Xue L., \\n2018, \\nChina, US   \n",
       "39     58                   Feinberg E., \\n2018, \\nChina, US   \n",
       "40     59                           Zhang L.,\\n2018, \\nChina   \n",
       "41     60                            Zhang B., 2018, \\nChina   \n",
       "42     64                             Hu B., \\n2018, \\nChina   \n",
       "43     65                            Le N., \\n2018, \\nTaiwan   \n",
       "44     66                             Sunseri J., 2018, \\nUS   \n",
       "45     67                             Cuperus J., 2018, \\nUS   \n",
       "46     68                             Li H., \\n2018, \\nChina   \n",
       "47     69                      Popova M., 2018, \\nRussia, US   \n",
       "48     72                           Hochuli J., \\n2018, \\nUS   \n",
       "49     73                        Khurana S., 2018, Qatar, US   \n",
       "50     74             Makrodimitris S., \\n2018,  Netherlands   \n",
       "51     75                                Fang C., 2018, \\nUS   \n",
       "52     76                             Adhikari B., 2017,\\nUS   \n",
       "53     77                     Armenteros J., 2017, \\nDenmark   \n",
       "54     78                        Jimenez J., \\n2017, \\nSpain   \n",
       "55     79                              Wang S., \\n2017, \\nUS   \n",
       "56     80                              Vang Y., \\n2017, \\nUS   \n",
       "57     81                       Müller A., 2017, Switzerland   \n",
       "58     82                         Szalkai B., 2017,\\nHungary   \n",
       "59     83                      Szalkai B., \\n2017, \\nHungary   \n",
       "60     86                           Yeh C., \\n2017, \\nUK, US   \n",
       "61     89              Simha R., 2015, \\nCanada, Germany, US   \n",
       "62     93                    Folkman L., \\n2014, \\nAustralia   \n",
       "63     94                                Li Z., \\n2014, \\nUS   \n",
       "64     99                    Eisenbeis S., \\n2012, \\nGermany   \n",
       "65    101                            Durrant J.,  2011, \\nUS   \n",
       "66    107                Briesemeister S., \\n2010, \\nGermany   \n",
       "67    120                         Shamim M., \\n2007, \\nIndia   \n",
       "68    123                   Zimmermann O., \\n2006, \\nGermany   \n",
       "69    126                        Capriotti E., 2005, \\nItaly   \n",
       "\n",
       "                                          7  \n",
       "0                                    Python  \n",
       "1                                    Python  \n",
       "2                                    Python  \n",
       "3                                    Python  \n",
       "4                      Matlab, Java, Python  \n",
       "5                                    Python  \n",
       "6                                    Python  \n",
       "7                                    Python  \n",
       "8                                    Python  \n",
       "9                                    Python  \n",
       "10                                   Python  \n",
       "11                            C++, \\nPython  \n",
       "12                                   Python  \n",
       "13                                   Python  \n",
       "14                                   Python  \n",
       "15                                   Python  \n",
       "16                                   Python  \n",
       "17                                   Python  \n",
       "18                                   Python  \n",
       "19                                   Python  \n",
       "20                                   Python  \n",
       "21                                   Python  \n",
       "22                                   Python  \n",
       "23                                   Python  \n",
       "24                                   Python  \n",
       "25                                   Python  \n",
       "26                                   Python  \n",
       "27                                   Python  \n",
       "28                                   Python  \n",
       "29                                   Python  \n",
       "30                                   Python  \n",
       "31                                   Python  \n",
       "32                   Python, Matlab,\\nLinux  \n",
       "33                           C++,  \\nPython  \n",
       "34                                   Python  \n",
       "35                           Java, \\nPython  \n",
       "36                                   Python  \n",
       "37                                   Python  \n",
       "38                                   Python  \n",
       "39                                   Python  \n",
       "40                                   Python  \n",
       "41                                   Python  \n",
       "42                         C, C++, \\nPython  \n",
       "43                                   Python  \n",
       "44                            C++, \\nPython  \n",
       "45                                   Python  \n",
       "46                                   Python  \n",
       "47                           Python, \\nCUDA  \n",
       "48                              C++, Python  \n",
       "49                        Python,\\n \\nLinux  \n",
       "50  Python, Matlab   Bioinformatics toolbox  \n",
       "51                                   Python  \n",
       "52                                   Python  \n",
       "53                                   Python  \n",
       "54                                   Python  \n",
       "55                                   Python  \n",
       "56                                   Python  \n",
       "57                                   Python  \n",
       "58                                   Python  \n",
       "59                                   Python  \n",
       "60                      Python, C++, Matlab  \n",
       "61                                   Python  \n",
       "62                            Python, Linux  \n",
       "63                           Python , Linux  \n",
       "64                                   Python  \n",
       "65                                   Python  \n",
       "66                      Python, Java, Linux  \n",
       "67     C++, Java , Python    Windows ,Linux  \n",
       "68               C, Python, Linux,  Windows  \n",
       "69                                   Python  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Esto busca la palabra Python en la columna de Programming Language, y nos da True y False cuando\n",
    "# encuentra la palabra y lo guarda en la variable Py.\n",
    "Py=df['7'].str.contains('Python')== True\n",
    "\n",
    "#esto nos da el data frame con los autores en donde esta la palabra python y lo asigna a DF_Py\n",
    "DF_Py=df[df['7'].str.contains('Python')== True].loc[:,['1','7']]\n",
    "          \n",
    "# esto nos da cuantos valores son para NaN, True y False para la palabra python y lo asigna a Py_Srs    \n",
    "Py_Srs=pd.Series(df['7']).str.contains('Python',regex=False).value_counts(dropna=False)    \n",
    "    \n",
    "# esto imprime los valores True y False para la palabra Python.\n",
    "print(Py.value_counts(),Py_Srs,DF_Py.shape,sep='\\n\\n')\n",
    "DF_Py.reset_index() # Python dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "saving-baltimore",
   "metadata": {
    "id": "saving-baltimore"
   },
   "source": [
    "---\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cf61e682",
   "metadata": {
    "id": "introductory-lightweight"
   },
   "source": [
    "# Lenguaje C++\n",
    "* En la celda de abajo buscamos en la columna de Programming Language todos los que contenga la palabra C++\n",
    "de esta forma para poder identificar cuantos artículos usar C++. Además, se muestran los autores de correspondencia.\n",
    "* Los resultado se compararon con la tabla de excely muestra los mismos resultados para el lenguaje C++."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "abandoned-courage",
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 492
    },
    "id": "abandoned-courage",
    "outputId": "cc029f68-293e-4262-ebd6-5b8c5a641f08"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False    80\n",
      "NaN      41\n",
      "True     12\n",
      "Name: 7, dtype: int64\n",
      "\n",
      "(12, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>1</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>19</td>\n",
       "      <td>Bond P., 2020, \\nUK</td>\n",
       "      <td>C++, \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>49</td>\n",
       "      <td>Pagès G., \\n2019, \\nFrance</td>\n",
       "      <td>C++,  \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>63</td>\n",
       "      <td>Frasca M., 2018, \\nItaly</td>\n",
       "      <td>C, C++, R, CUDA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>64</td>\n",
       "      <td>Hu B., \\n2018, \\nChina</td>\n",
       "      <td>C, C++, \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>66</td>\n",
       "      <td>Sunseri J., 2018, \\nUS</td>\n",
       "      <td>C++, \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>72</td>\n",
       "      <td>Hochuli J., \\n2018, \\nUS</td>\n",
       "      <td>C++, Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>85</td>\n",
       "      <td>Ragoza M., 2017,\\nUS</td>\n",
       "      <td>C++</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>86</td>\n",
       "      <td>Yeh C., \\n2017, \\nUK, US</td>\n",
       "      <td>Python, C++, Matlab</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>111</td>\n",
       "      <td>Hong E.,\\n2009, \\nUS</td>\n",
       "      <td>C++, Linux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>117</td>\n",
       "      <td>Katzman S., 2008, \\nUS</td>\n",
       "      <td>C++</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>119</td>\n",
       "      <td>Armstrong K., \\n2008, \\nUS</td>\n",
       "      <td>C++</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>120</td>\n",
       "      <td>Shamim M., \\n2007, \\nIndia</td>\n",
       "      <td>C++, Java , Python    Windows ,Linux</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    index                           1                                     7\n",
       "0      19         Bond P., 2020, \\nUK                         C++, \\nPython\n",
       "1      49  Pagès G., \\n2019, \\nFrance                        C++,  \\nPython\n",
       "2      63    Frasca M., 2018, \\nItaly                       C, C++, R, CUDA\n",
       "3      64      Hu B., \\n2018, \\nChina                      C, C++, \\nPython\n",
       "4      66      Sunseri J., 2018, \\nUS                         C++, \\nPython\n",
       "5      72    Hochuli J., \\n2018, \\nUS                           C++, Python\n",
       "6      85        Ragoza M., 2017,\\nUS                                   C++\n",
       "7      86    Yeh C., \\n2017, \\nUK, US                   Python, C++, Matlab\n",
       "8     111        Hong E.,\\n2009, \\nUS                            C++, Linux\n",
       "9     117      Katzman S., 2008, \\nUS                                   C++\n",
       "10    119  Armstrong K., \\n2008, \\nUS                                   C++\n",
       "11    120  Shamim M., \\n2007, \\nIndia  C++, Java , Python    Windows ,Linux"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Esto busca la palabra C++ en la columna de Programming Language, y nos da True, False y NaN\n",
    "\n",
    "#esto nos da el data frame con los autores en donde esta la palabra c++ y lo asigna a DF_cpp\n",
    "DF_Cpp=df[df['7'].str.contains('C++',regex=False)== True].loc[:,['1','7']]\n",
    "          \n",
    "# esto nos da cuantos valores son para NaN, True y False para la palabra c++ y lo asigna a Cpp_Srs    \n",
    "Cpp_Srs=pd.Series(df['7']).str.contains('C++',regex=False).value_counts(dropna=False)    \n",
    "    \n",
    "# esto imprime los valores True y False para el lenguaje\n",
    "print(Cpp_Srs,DF_Cpp.shape,sep='\\n\\n') \n",
    "DF_Cpp.reset_index() #  dataframe\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92a6441f",
   "metadata": {
    "id": "0FQtKT8EXBsT"
   },
   "source": [
    "---\n",
    "---\n",
    "# Lenguaje C\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "qhs6DDn1XPvY",
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 647
    },
    "id": "qhs6DDn1XPvY",
    "outputId": "8a296754-73e2-45bc-f191-9a3717323203"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False    75\n",
      "NaN      41\n",
      "True     17\n",
      "Name: 7, dtype: int64\n",
      "\n",
      "(17, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>1</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "      <td>Feger G., 2020,\\nCzech Republic, France</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>19</td>\n",
       "      <td>Bond P., 2020, \\nUK</td>\n",
       "      <td>C++, \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>49</td>\n",
       "      <td>Pagès G., \\n2019, \\nFrance</td>\n",
       "      <td>C++,  \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>63</td>\n",
       "      <td>Frasca M., 2018, \\nItaly</td>\n",
       "      <td>C, C++, R, CUDA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>64</td>\n",
       "      <td>Hu B., \\n2018, \\nChina</td>\n",
       "      <td>C, C++, \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>66</td>\n",
       "      <td>Sunseri J., 2018, \\nUS</td>\n",
       "      <td>C++, \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>69</td>\n",
       "      <td>Popova M., 2018, \\nRussia, US</td>\n",
       "      <td>Python, \\nCUDA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>72</td>\n",
       "      <td>Hochuli J., \\n2018, \\nUS</td>\n",
       "      <td>C++, Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>85</td>\n",
       "      <td>Ragoza M., 2017,\\nUS</td>\n",
       "      <td>C++</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>86</td>\n",
       "      <td>Yeh C., \\n2017, \\nUK, US</td>\n",
       "      <td>Python, C++, Matlab</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>98</td>\n",
       "      <td>Qi Y., \\n2012, \\nUS</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>111</td>\n",
       "      <td>Hong E.,\\n2009, \\nUS</td>\n",
       "      <td>C++, Linux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>117</td>\n",
       "      <td>Katzman S., 2008, \\nUS</td>\n",
       "      <td>C++</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>119</td>\n",
       "      <td>Armstrong K., \\n2008, \\nUS</td>\n",
       "      <td>C++</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>120</td>\n",
       "      <td>Shamim M., \\n2007, \\nIndia</td>\n",
       "      <td>C++, Java , Python    Windows ,Linux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>123</td>\n",
       "      <td>Zimmermann O., \\n2006, \\nGermany</td>\n",
       "      <td>C, Python, Linux,  Windows</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>132</td>\n",
       "      <td>Ladunga I., \\n1991, \\nHungary</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    index                                        1  \\\n",
       "0      17  Feger G., 2020,\\nCzech Republic, France   \n",
       "1      19                      Bond P., 2020, \\nUK   \n",
       "2      49               Pagès G., \\n2019, \\nFrance   \n",
       "3      63                 Frasca M., 2018, \\nItaly   \n",
       "4      64                   Hu B., \\n2018, \\nChina   \n",
       "5      66                   Sunseri J., 2018, \\nUS   \n",
       "6      69            Popova M., 2018, \\nRussia, US   \n",
       "7      72                 Hochuli J., \\n2018, \\nUS   \n",
       "8      85                     Ragoza M., 2017,\\nUS   \n",
       "9      86                 Yeh C., \\n2017, \\nUK, US   \n",
       "10     98                      Qi Y., \\n2012, \\nUS   \n",
       "11    111                     Hong E.,\\n2009, \\nUS   \n",
       "12    117                   Katzman S., 2008, \\nUS   \n",
       "13    119               Armstrong K., \\n2008, \\nUS   \n",
       "14    120               Shamim M., \\n2007, \\nIndia   \n",
       "15    123         Zimmermann O., \\n2006, \\nGermany   \n",
       "16    132            Ladunga I., \\n1991, \\nHungary   \n",
       "\n",
       "                                       7  \n",
       "0                                      C  \n",
       "1                          C++, \\nPython  \n",
       "2                         C++,  \\nPython  \n",
       "3                        C, C++, R, CUDA  \n",
       "4                       C, C++, \\nPython  \n",
       "5                          C++, \\nPython  \n",
       "6                         Python, \\nCUDA  \n",
       "7                            C++, Python  \n",
       "8                                    C++  \n",
       "9                    Python, C++, Matlab  \n",
       "10                                     C  \n",
       "11                            C++, Linux  \n",
       "12                                   C++  \n",
       "13                                   C++  \n",
       "14  C++, Java , Python    Windows ,Linux  \n",
       "15            C, Python, Linux,  Windows  \n",
       "16                                     C  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Esto busca la palabra C++ en la columna de Programming Language, y nos da True, False y NaN\n",
    "\n",
    "#esto nos da el data frame con los autores en donde esta la letra C y lo asigna a DF_cpp\n",
    "DF_C=df[df['7'].str.contains('C',regex=False)== True].loc[:,['1','7']]\n",
    "          \n",
    "# esto nos da cuantos valores son para NaN, True y False para la palabra c++ y lo asigna a Cpp_Srs    \n",
    "C_Srs=pd.Series(df['7']).str.contains('C',regex=False).value_counts(dropna=False)    \n",
    "    \n",
    "# esto imprime los valores True y False para el lenguaje\n",
    "print(C_Srs,DF_C.shape,sep='\\n\\n')\n",
    "DF_C.reset_index() # dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3dd7599c",
   "metadata": {
    "id": "enz_COP2TAbW"
   },
   "source": [
    "# Lenguaje R "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "banner-syndicate",
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 142
    },
    "id": "banner-syndicate",
    "outputId": "31eebffc-b6d8-429e-b179-55e0905a7a73"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False    89\n",
      "NaN      41\n",
      "True      3\n",
      "Name: 7, dtype: int64\n",
      "\n",
      "(3, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>1</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>62</td>\n",
       "      <td>Long H.,  \\n2018, \\nChina, US</td>\n",
       "      <td>R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>63</td>\n",
       "      <td>Frasca M., 2018, \\nItaly</td>\n",
       "      <td>C, C++, R, CUDA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>109</td>\n",
       "      <td>Tian J., \\n2010, \\nChina, US</td>\n",
       "      <td>R, Perl,  Linux</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   index                              1                7\n",
       "0     62  Long H.,  \\n2018, \\nChina, US                R\n",
       "1     63       Frasca M., 2018, \\nItaly  C, C++, R, CUDA\n",
       "2    109   Tian J., \\n2010, \\nChina, US  R, Perl,  Linux"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Esto busca la palabra R en la columna de Programming Language, y nos da True, False y NaN\n",
    "\n",
    "#esto nos da el data frame con los autores en donde esta la letra R y lo asigna a DF_Srs   \n",
    "RR_Srs=pd.Series(df['7']).str.contains('R',regex=False).value_counts(dropna=False)\n",
    "\n",
    "#esto nos da el data frame con los autores en donde esta la letra R\n",
    "DF_RR=df[df['7'].str.contains('R',regex=False)== True].loc[:,['1','7']]\n",
    "\n",
    "# esto imprime los valores True y False para la palabra del lenguaje \n",
    "print(RR_Srs,DF_RR.shape,sep='\\n\\n')\n",
    "DF_RR.reset_index() # dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1c590644",
   "metadata": {},
   "source": [
    "## Lenguaje Perl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2f9ec1d1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False    89\n",
      "NaN      41\n",
      "True      3\n",
      "Name: 7, dtype: int64\n",
      "\n",
      "(3, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>1</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>97</td>\n",
       "      <td>Traoré S., 2013, \\nFrance</td>\n",
       "      <td>Perl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>103</td>\n",
       "      <td>Ashkenazy H., \\n2011, \\nIsrael</td>\n",
       "      <td>Perl</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>109</td>\n",
       "      <td>Tian J., \\n2010, \\nChina, US</td>\n",
       "      <td>R, Perl,  Linux</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   index                               1                7\n",
       "0     97       Traoré S., 2013, \\nFrance             Perl\n",
       "1    103  Ashkenazy H., \\n2011, \\nIsrael             Perl\n",
       "2    109    Tian J., \\n2010, \\nChina, US  R, Perl,  Linux"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Esto busca la palabra PERL en la columna de Programming Language, y nos da True, False y NaN\n",
    "\n",
    "#esto nos da el data frame con los autores en donde esta la letra R y lo asigna a DF_Srs   \n",
    "PERL_Srs=pd.Series(df['7']).str.contains('Perl',regex=False).value_counts(dropna=False)\n",
    "\n",
    "\n",
    "#esto nos da el data frame con los autores en donde esta la letra R\n",
    "DF_PERL=df[df['7'].str.contains('Perl',regex=False)== True].loc[:,['1','7']]\n",
    "\n",
    "\n",
    "# esto imprime los valores True y False para la palabra del lenguaje \n",
    "print(PERL_Srs,DF_PERL.shape,sep='\\n\\n')\n",
    "DF_PERL.reset_index() # dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0ec483bc",
   "metadata": {},
   "source": [
    "# Lenguaje Java"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "857b831a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False    85\n",
      "NaN      41\n",
      "True      7\n",
      "Name: 7, dtype: int64\n",
      "\n",
      "(7, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>1</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6</td>\n",
       "      <td>Wang M.., 2020, US</td>\n",
       "      <td>Matlab, Java, Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>51</td>\n",
       "      <td>Yu C., \\n2019, \\nUS</td>\n",
       "      <td>Java, \\nPython</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>90</td>\n",
       "      <td>Khan Z., \\n2014, \\nPakistan</td>\n",
       "      <td>Java, Matlab</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>107</td>\n",
       "      <td>Briesemeister S., \\n2010, \\nGermany</td>\n",
       "      <td>Python, Java, Linux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>108</td>\n",
       "      <td>Lin G., \\n2010, \\nUS</td>\n",
       "      <td>Java</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>113</td>\n",
       "      <td>Ebrahimpour A., \\n2008,  Malaysia</td>\n",
       "      <td>Java</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>120</td>\n",
       "      <td>Shamim M., \\n2007, \\nIndia</td>\n",
       "      <td>C++, Java , Python    Windows ,Linux</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   index                                    1  \\\n",
       "0      6                   Wang M.., 2020, US   \n",
       "1     51                  Yu C., \\n2019, \\nUS   \n",
       "2     90          Khan Z., \\n2014, \\nPakistan   \n",
       "3    107  Briesemeister S., \\n2010, \\nGermany   \n",
       "4    108                 Lin G., \\n2010, \\nUS   \n",
       "5    113    Ebrahimpour A., \\n2008,  Malaysia   \n",
       "6    120           Shamim M., \\n2007, \\nIndia   \n",
       "\n",
       "                                      7  \n",
       "0                  Matlab, Java, Python  \n",
       "1                        Java, \\nPython  \n",
       "2                          Java, Matlab  \n",
       "3                   Python, Java, Linux  \n",
       "4                                  Java  \n",
       "5                                  Java  \n",
       "6  C++, Java , Python    Windows ,Linux  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Esto busca la palabra Java en la columna de Programming Language, y nos da True, False y NaN\n",
    "\n",
    "java_Srs=pd.Series(df['7']).str.contains('Java',regex=False).value_counts(dropna=False)\n",
    "\n",
    "DF_java=df[df['7'].str.contains('Java',regex=False)== True].loc[:,['1','7']]\n",
    "\n",
    "# esto imprime los valores True y False para la palabra del lenguaje \n",
    "print(java_Srs,DF_java.shape,sep='\\n\\n')\n",
    "DF_java.reset_index() # dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0dbd171",
   "metadata": {},
   "source": [
    "# Lenguaje BASH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0b8c5b00",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False    91\n",
      "NaN      41\n",
      "True      1\n",
      "Name: 7, dtype: int64\n",
      "\n",
      "(1, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>1</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>102</td>\n",
       "      <td>Ebina T.,  \\n2011, \\nJapan</td>\n",
       "      <td>Bash script</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   index                           1            7\n",
       "0    102  Ebina T.,  \\n2011, \\nJapan  Bash script"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Esto busca la palabra Bash en la columna de Programming Language, y nos da True, False y NaN\n",
    "  \n",
    "bash_Srs=pd.Series(df['7']).str.contains('Bash',regex=False).value_counts(dropna=False)\n",
    "\n",
    "\n",
    "DF_bash=df[df['7'].str.contains('Bash',regex=False)== True].loc[:,['1','7']]\n",
    "\n",
    "\n",
    "# esto imprime los valores True y False para la palabra del lenguaje \n",
    "print(bash_Srs,DF_bash.shape,sep='\\n\\n')\n",
    "DF_bash.reset_index() # dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f65c71b2",
   "metadata": {},
   "source": [
    "# Lenguaje MATLAB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4ff23287",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False    84\n",
      "NaN      41\n",
      "True      8\n",
      "Name: 7, dtype: int64\n",
      "\n",
      "(8, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>1</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6</td>\n",
       "      <td>Wang M.., 2020, US</td>\n",
       "      <td>Matlab, Java, Python</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>24</td>\n",
       "      <td>Lin J., 2019, China</td>\n",
       "      <td>Matlab</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>48</td>\n",
       "      <td>Fu H.,\\n2019, \\nChina</td>\n",
       "      <td>Python, Matlab,\\nLinux</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>74</td>\n",
       "      <td>Makrodimitris S., \\n2018,  Netherlands</td>\n",
       "      <td>Python, Matlab   Bioinformatics toolbox</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>86</td>\n",
       "      <td>Yeh C., \\n2017, \\nUK, US</td>\n",
       "      <td>Python, C++, Matlab</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>90</td>\n",
       "      <td>Khan Z., \\n2014, \\nPakistan</td>\n",
       "      <td>Java, Matlab</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>121</td>\n",
       "      <td>Liao J., \\n2007, \\nUS</td>\n",
       "      <td>Matlab</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>122</td>\n",
       "      <td>Raveh B., \\n2007, \\nIsrael</td>\n",
       "      <td>Matlab</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   index                                       1  \\\n",
       "0      6                      Wang M.., 2020, US   \n",
       "1     24                     Lin J., 2019, China   \n",
       "2     48                   Fu H.,\\n2019, \\nChina   \n",
       "3     74  Makrodimitris S., \\n2018,  Netherlands   \n",
       "4     86                Yeh C., \\n2017, \\nUK, US   \n",
       "5     90             Khan Z., \\n2014, \\nPakistan   \n",
       "6    121                   Liao J., \\n2007, \\nUS   \n",
       "7    122              Raveh B., \\n2007, \\nIsrael   \n",
       "\n",
       "                                         7  \n",
       "0                     Matlab, Java, Python  \n",
       "1                                   Matlab  \n",
       "2                   Python, Matlab,\\nLinux  \n",
       "3  Python, Matlab   Bioinformatics toolbox  \n",
       "4                      Python, C++, Matlab  \n",
       "5                             Java, Matlab  \n",
       "6                                   Matlab  \n",
       "7                                   Matlab  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Esto busca la palabra MATLAB en la columna de Programming Language, y nos da True, False y NaN\n",
    "  \n",
    "MAT_Srs=pd.Series(df['7']).str.contains('Matlab',regex=False).value_counts(dropna=False)\n",
    "\n",
    "\n",
    "DF_MAT=df[df['7'].str.contains('Matlab',regex=False)== True].loc[:,['1','7']]\n",
    "\n",
    "\n",
    "# esto imprime los valores True y False para la palabra del lenguaje \n",
    "print(MAT_Srs,DF_MAT.shape,sep='\\n\\n')\n",
    "DF_MAT.reset_index() # dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd1f3aa7",
   "metadata": {},
   "source": [
    "# Shell script"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "20a87dcc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False    91\n",
      "NaN      41\n",
      "True      1\n",
      "Name: 7, dtype: int64\n",
      "\n",
      "(1, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>1</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>104</td>\n",
       "      <td>Yang Y.,\\n2011, \\nUS</td>\n",
       "      <td>Shell script</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   index                     1             7\n",
       "0    104  Yang Y.,\\n2011, \\nUS  Shell script"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Esto busca la palabra shell en la columna de Programming Language, y nos da True, False y NaN\n",
    "  \n",
    "Shell_Srs=pd.Series(df['7']).str.contains('Shell',regex=False).value_counts(dropna=False)\n",
    "\n",
    "DF_Shell=df[df['7'].str.contains('Shell',regex=False)== True].loc[:,['1','7']]\n",
    "\n",
    "# esto imprime los valores True y False para la palabra del lenguaje \n",
    "print(Shell_Srs,DF_Shell.shape,sep='\\n\\n')\n",
    "DF_Shell.reset_index() # dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "199ecf85",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Shell_Srs[2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "through-vessel",
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 316
    },
    "id": "through-vessel",
    "outputId": "fef7688c-870f-4876-f985-89cffd4ce854"
   },
   "outputs": [],
   "source": [
    "plt.style.use('ggplot')\n",
    "\n",
    "x = ['Python', 'C++', 'C', 'Java', 'R', 'Java','MATLAB','Perl','Bash','CUDA']\n",
    "values = [29,   16,    5,     3,    3,    3,      2,      2,      2,    1]\n",
    "\n",
    "x_pos = [i for i, _ in enumerate(x)]\n",
    "\n",
    "# plt.figure(figsize = (6,6), dpi=80)\n",
    "plt.bar(x_pos, values, width=0.4,color='slateblue',edgecolor='white')\n",
    "plt.xlabel(\"Programming Languages\")\n",
    "plt.ylabel(\"# of Articles\")\n",
    "plt.title(\"Programming Language Usage\")\n",
    "plt.ylim(0, 26)\n",
    "plt.xticks(x_pos, x)\n",
    "plt.grid(color='black', linestyle='-', linewidth=0.2)\n",
    "plt.show()\n",
    "plt.savefig('/content/drive/MyDrive/Colab Notebooks/Lenguajes.png', dpi=150)\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "colab": {
   "collapsed_sections": [],
   "name": "Pandas_Manuscrito_Tabla_Anpalisis.ipynb",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
